// @aws-sdk/client-sqs/dist-cjs/index.js
const { awsEndpointFunctions, emitWarningIfUnsupportedVersion: emitWarningIfUnsupportedVersion$1, createDefaultUserAgentProvider, NODE_APP_ID_CONFIG_OPTIONS, getAwsRegionExtensionConfiguration, resolveAwsRegionExtensionConfiguration, resolveUserAgentConfig, resolveHostHeaderConfig, getUserAgentPlugin, getHostHeaderPlugin, getLoggerPlugin, getRecursionDetectionPlugin, getLongPollPlugin } = require("@aws-sdk/core/client");
const { resolveQueueUrlConfig, getQueueUrlPlugin, getReceiveMessagePlugin, getSendMessageBatchPlugin, getSendMessagePlugin } = require("@aws-sdk/middleware-sdk-sqs");
const { getHttpAuthSchemeEndpointRuleSetPlugin, DefaultIdentityProviderConfig, getHttpSigningPlugin, createPaginator } = require("@smithy/core");
const { normalizeProvider, getSmithyContext, ServiceException, NoOpLogger, emitWarningIfUnsupportedVersion, loadConfigsForDefaultMode, getDefaultExtensionConfiguration, resolveDefaultRuntimeConfig, Client, makeBuilder, createAggregatedClient } = require("@smithy/core/client");
const { Command: $Command } = require("@smithy/core/client");
exports.$Command = $Command;
exports.__Client = Client;
const { resolveDefaultsModeConfig, loadConfig, NODE_USE_FIPS_ENDPOINT_CONFIG_OPTIONS, NODE_USE_DUALSTACK_ENDPOINT_CONFIG_OPTIONS, NODE_REGION_CONFIG_OPTIONS, NODE_REGION_CONFIG_FILE_OPTIONS, resolveRegionConfig } = require("@smithy/core/config");
const { BinaryDecisionDiagram, EndpointCache, decideEndpoint, customEndpointFunctions, resolveEndpointConfig, getEndpointPlugin } = require("@smithy/core/endpoints");
const { parseUrl, getHttpHandlerExtensionConfiguration, resolveHttpHandlerRuntimeConfig, getContentLengthPlugin } = require("@smithy/core/protocols");
const { DEFAULT_RETRY_MODE, NODE_RETRY_MODE_CONFIG_OPTIONS, NODE_MAX_ATTEMPT_CONFIG_OPTIONS, resolveRetryConfig, getRetryPlugin } = require("@smithy/core/retry");
const { TypeRegistry, getSchemaSerdePlugin } = require("@smithy/core/schema");
const { resolveAwsSdkSigV4Config, AwsSdkSigV4Signer, NODE_AUTH_SCHEME_PREFERENCE_OPTIONS } = require("@aws-sdk/core/httpAuthSchemes");
const { defaultProvider } = require("@aws-sdk/credential-provider-node");
const { toUtf8, fromUtf8, toBase64, fromBase64, calculateBodyLength } = require("@smithy/core/serde");
const { streamCollector, NodeHttpHandler } = require("@smithy/node-http-handler");
const { AwsJson1_0Protocol } = require("@aws-sdk/core/protocols");
const { Sha256, Md5 } = require("@smithy/core/checksum");

const defaultSQSHttpAuthSchemeParametersProvider = async (config, context, input) => {
    return {
        operation: getSmithyContext(context).operation,
        region: await normalizeProvider(config.region)() || (() => {
            throw new Error("expected `region` to be configured for `aws.auth#sigv4`");
        })(),
    };
};
function createAwsAuthSigv4HttpAuthOption(authParameters) {
    return {
        schemeId: "aws.auth#sigv4",
        signingProperties: {
            name: "sqs",
            region: authParameters.region,
        },
        propertiesExtractor: (config, context) => ({
            signingProperties: {
                config,
                context,
            },
        }),
    };
}
const defaultSQSHttpAuthSchemeProvider = (authParameters) => {
    const options = [];
    switch (authParameters.operation) {
        default: {
            options.push(createAwsAuthSigv4HttpAuthOption(authParameters));
        }
    }
    return options;
};
const resolveHttpAuthSchemeConfig = (config) => {
    const config_0 = resolveAwsSdkSigV4Config(config);
    return Object.assign(config_0, {
        authSchemePreference: normalizeProvider(config.authSchemePreference ?? []),
    });
};

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "sqs",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

var version = "3.1084.0";
var packageInfo = {
	version: version};

const k = "ref";
const a = -1, b = true, c = "isSet", d = "PartitionResult", e = "booleanEquals", f = "getAttr", g = { [k]: "Endpoint" }, h = { [k]: d }, i = {}, j = [{ [k]: "Region" }];
const _data = {
    conditions: [
        [c, [g]],
        [c, j],
        ["aws.partition", j, d],
        [e, [{ [k]: "UseFIPS" }, b]],
        [e, [{ [k]: "UseDualStack" }, b]],
        [e, [{ fn: f, argv: [h, "supportsDualStack"] }, b]],
        [e, [{ fn: f, argv: [h, "supportsFIPS"] }, b]],
        ["stringEquals", [{ fn: f, argv: [h, "name"] }, "aws-us-gov"]]
    ],
    results: [
        [a],
        [a, "Invalid Configuration: FIPS and custom endpoint are not supported"],
        [a, "Invalid Configuration: Dualstack and custom endpoint are not supported"],
        [g, i],
        ["https://sqs-fips.{Region}.{PartitionResult#dualStackDnsSuffix}", i],
        [a, "FIPS and DualStack are enabled, but this partition does not support one or both"],
        ["https://sqs.{Region}.amazonaws.com", i],
        ["https://sqs-fips.{Region}.{PartitionResult#dnsSuffix}", i],
        [a, "FIPS is enabled but this partition does not support FIPS"],
        ["https://sqs.{Region}.{PartitionResult#dualStackDnsSuffix}", i],
        [a, "DualStack is enabled but this partition does not support DualStack"],
        ["https://sqs.{Region}.{PartitionResult#dnsSuffix}", i],
        [a, "Invalid Configuration: Missing Region"]
    ]
};
const root = 2;
const r = 100_000_000;
const nodes = new Int32Array([
    -1, 1, -1,
    0, 13, 3,
    1, 4, r + 12,
    2, 5, r + 12,
    3, 8, 6,
    4, 7, r + 11,
    5, r + 9, r + 10,
    4, 11, 9,
    6, 10, r + 8,
    7, r + 6, r + 7,
    5, 12, r + 5,
    6, r + 4, r + 5,
    3, r + 1, 14,
    4, r + 2, r + 3,
]);
const bdd = BinaryDecisionDiagram.from(nodes, root, _data.conditions, _data.results);

const cache = new EndpointCache({
    size: 50,
    params: ["Endpoint", "Region", "UseDualStack", "UseFIPS"],
});
const defaultEndpointResolver = (endpointParams, context = {}) => {
    return cache.get(endpointParams, () => decideEndpoint(bdd, {
        endpointParams: endpointParams,
        logger: context.logger,
    }));
};
customEndpointFunctions.aws = awsEndpointFunctions;

class SQSServiceException extends ServiceException {
    constructor(options) {
        super(options);
        Object.setPrototypeOf(this, SQSServiceException.prototype);
    }
}

class InvalidAddress extends SQSServiceException {
    name = "InvalidAddress";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidAddress",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidAddress.prototype);
    }
}
class InvalidSecurity extends SQSServiceException {
    name = "InvalidSecurity";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidSecurity",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidSecurity.prototype);
    }
}
class OverLimit extends SQSServiceException {
    name = "OverLimit";
    $fault = "client";
    constructor(opts) {
        super({
            name: "OverLimit",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, OverLimit.prototype);
    }
}
class QueueDoesNotExist extends SQSServiceException {
    name = "QueueDoesNotExist";
    $fault = "client";
    constructor(opts) {
        super({
            name: "QueueDoesNotExist",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, QueueDoesNotExist.prototype);
    }
}
class RequestThrottled extends SQSServiceException {
    name = "RequestThrottled";
    $fault = "client";
    constructor(opts) {
        super({
            name: "RequestThrottled",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, RequestThrottled.prototype);
    }
}
class UnsupportedOperation extends SQSServiceException {
    name = "UnsupportedOperation";
    $fault = "client";
    constructor(opts) {
        super({
            name: "UnsupportedOperation",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, UnsupportedOperation.prototype);
    }
}
class ResourceNotFoundException extends SQSServiceException {
    name = "ResourceNotFoundException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
    }
}
class MessageNotInflight extends SQSServiceException {
    name = "MessageNotInflight";
    $fault = "client";
    constructor(opts) {
        super({
            name: "MessageNotInflight",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, MessageNotInflight.prototype);
    }
}
class ReceiptHandleIsInvalid extends SQSServiceException {
    name = "ReceiptHandleIsInvalid";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ReceiptHandleIsInvalid",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ReceiptHandleIsInvalid.prototype);
    }
}
class BatchEntryIdsNotDistinct extends SQSServiceException {
    name = "BatchEntryIdsNotDistinct";
    $fault = "client";
    constructor(opts) {
        super({
            name: "BatchEntryIdsNotDistinct",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, BatchEntryIdsNotDistinct.prototype);
    }
}
class EmptyBatchRequest extends SQSServiceException {
    name = "EmptyBatchRequest";
    $fault = "client";
    constructor(opts) {
        super({
            name: "EmptyBatchRequest",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, EmptyBatchRequest.prototype);
    }
}
class InvalidBatchEntryId extends SQSServiceException {
    name = "InvalidBatchEntryId";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidBatchEntryId",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidBatchEntryId.prototype);
    }
}
class TooManyEntriesInBatchRequest extends SQSServiceException {
    name = "TooManyEntriesInBatchRequest";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TooManyEntriesInBatchRequest",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyEntriesInBatchRequest.prototype);
    }
}
class InvalidAttributeName extends SQSServiceException {
    name = "InvalidAttributeName";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidAttributeName",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidAttributeName.prototype);
    }
}
class InvalidAttributeValue extends SQSServiceException {
    name = "InvalidAttributeValue";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidAttributeValue",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidAttributeValue.prototype);
    }
}
class QueueDeletedRecently extends SQSServiceException {
    name = "QueueDeletedRecently";
    $fault = "client";
    constructor(opts) {
        super({
            name: "QueueDeletedRecently",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, QueueDeletedRecently.prototype);
    }
}
class QueueNameExists extends SQSServiceException {
    name = "QueueNameExists";
    $fault = "client";
    constructor(opts) {
        super({
            name: "QueueNameExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, QueueNameExists.prototype);
    }
}
class InvalidIdFormat extends SQSServiceException {
    name = "InvalidIdFormat";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidIdFormat",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidIdFormat.prototype);
    }
}
class PurgeQueueInProgress extends SQSServiceException {
    name = "PurgeQueueInProgress";
    $fault = "client";
    constructor(opts) {
        super({
            name: "PurgeQueueInProgress",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, PurgeQueueInProgress.prototype);
    }
}
class KmsAccessDenied extends SQSServiceException {
    name = "KmsAccessDenied";
    $fault = "client";
    constructor(opts) {
        super({
            name: "KmsAccessDenied",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, KmsAccessDenied.prototype);
    }
}
class KmsDisabled extends SQSServiceException {
    name = "KmsDisabled";
    $fault = "client";
    constructor(opts) {
        super({
            name: "KmsDisabled",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, KmsDisabled.prototype);
    }
}
class KmsInvalidKeyUsage extends SQSServiceException {
    name = "KmsInvalidKeyUsage";
    $fault = "client";
    constructor(opts) {
        super({
            name: "KmsInvalidKeyUsage",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, KmsInvalidKeyUsage.prototype);
    }
}
class KmsInvalidState extends SQSServiceException {
    name = "KmsInvalidState";
    $fault = "client";
    constructor(opts) {
        super({
            name: "KmsInvalidState",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, KmsInvalidState.prototype);
    }
}
class KmsNotFound extends SQSServiceException {
    name = "KmsNotFound";
    $fault = "client";
    constructor(opts) {
        super({
            name: "KmsNotFound",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, KmsNotFound.prototype);
    }
}
class KmsOptInRequired extends SQSServiceException {
    name = "KmsOptInRequired";
    $fault = "client";
    constructor(opts) {
        super({
            name: "KmsOptInRequired",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, KmsOptInRequired.prototype);
    }
}
class KmsThrottled extends SQSServiceException {
    name = "KmsThrottled";
    $fault = "client";
    constructor(opts) {
        super({
            name: "KmsThrottled",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, KmsThrottled.prototype);
    }
}
class InvalidMessageContents extends SQSServiceException {
    name = "InvalidMessageContents";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidMessageContents",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidMessageContents.prototype);
    }
}
class BatchRequestTooLong extends SQSServiceException {
    name = "BatchRequestTooLong";
    $fault = "client";
    constructor(opts) {
        super({
            name: "BatchRequestTooLong",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, BatchRequestTooLong.prototype);
    }
}

const _A = "Actions";
const _AN = "ActionName";
const _ANOMM = "ApproximateNumberOfMessagesMoved";
const _ANOMTM = "ApproximateNumberOfMessagesToMove";
const _ANt = "AttributeNames";
const _ANtt = "AttributeName";
const _AP = "AddPermission";
const _APR = "AddPermissionRequest";
const _AWSAI = "AWSAccountIds";
const _AWSAIc = "AWSAccountId";
const _At = "Attributes";
const _Att = "Attribute";
const _B = "Body";
const _BEIND = "BatchEntryIdsNotDistinct";
const _BL = "BinaryList";
const _BLV = "BinaryListValues";
const _BLVi = "BinaryListValue";
const _BREE = "BatchResultErrorEntry";
const _BREEL = "BatchResultErrorEntryList";
const _BRTL = "BatchRequestTooLong";
const _BV = "BinaryValue";
const _C = "Code";
const _CMMT = "CancelMessageMoveTask";
const _CMMTR = "CancelMessageMoveTaskRequest";
const _CMMTRa = "CancelMessageMoveTaskResult";
const _CMV = "ChangeMessageVisibility";
const _CMVB = "ChangeMessageVisibilityBatch";
const _CMVBR = "ChangeMessageVisibilityBatchRequest";
const _CMVBRE = "ChangeMessageVisibilityBatchRequestEntry";
const _CMVBREL = "ChangeMessageVisibilityBatchRequestEntryList";
const _CMVBRELh = "ChangeMessageVisibilityBatchResultEntryList";
const _CMVBREh = "ChangeMessageVisibilityBatchResultEntry";
const _CMVBRh = "ChangeMessageVisibilityBatchResult";
const _CMVR = "ChangeMessageVisibilityRequest";
const _CQ = "CreateQueue";
const _CQR = "CreateQueueRequest";
const _CQRr = "CreateQueueResult";
const _DA = "DestinationArn";
const _DM = "DeleteMessage";
const _DMB = "DeleteMessageBatch";
const _DMBR = "DeleteMessageBatchRequest";
const _DMBRE = "DeleteMessageBatchRequestEntry";
const _DMBREL = "DeleteMessageBatchRequestEntryList";
const _DMBRELe = "DeleteMessageBatchResultEntryList";
const _DMBREe = "DeleteMessageBatchResultEntry";
const _DMBRe = "DeleteMessageBatchResult";
const _DMR = "DeleteMessageRequest";
const _DQ = "DeleteQueue";
const _DQR = "DeleteQueueRequest";
const _DS = "DelaySeconds";
const _DT = "DataType";
const _E = "Entries";
const _EBR = "EmptyBatchRequest";
const _F = "Failed";
const _FR = "FailureReason";
const _GQA = "GetQueueAttributes";
const _GQAR = "GetQueueAttributesRequest";
const _GQARe = "GetQueueAttributesResult";
const _GQU = "GetQueueUrl";
const _GQUR = "GetQueueUrlRequest";
const _GQURe = "GetQueueUrlResult";
const _I = "Id";
const _IA = "InvalidAddress";
const _IAN = "InvalidAttributeName";
const _IAV = "InvalidAttributeValue";
const _IBEI = "InvalidBatchEntryId";
const _IIF = "InvalidIdFormat";
const _IMC = "InvalidMessageContents";
const _IS = "InvalidSecurity";
const _K = "Key";
const _KAD = "KmsAccessDenied";
const _KD = "KmsDisabled";
const _KIKU = "KmsInvalidKeyUsage";
const _KIS = "KmsInvalidState";
const _KNF = "KmsNotFound";
const _KOIR = "KmsOptInRequired";
const _KT = "KmsThrottled";
const _L = "Label";
const _LDLSQ = "ListDeadLetterSourceQueues";
const _LDLSQR = "ListDeadLetterSourceQueuesRequest";
const _LDLSQRi = "ListDeadLetterSourceQueuesResult";
const _LMMT = "ListMessageMoveTasks";
const _LMMTR = "ListMessageMoveTasksRequest";
const _LMMTRE = "ListMessageMoveTasksResultEntry";
const _LMMTREL = "ListMessageMoveTasksResultEntryList";
const _LMMTRi = "ListMessageMoveTasksResult";
const _LQ = "ListQueues";
const _LQR = "ListQueuesRequest";
const _LQRi = "ListQueuesResult";
const _LQT = "ListQueueTags";
const _LQTR = "ListQueueTagsRequest";
const _LQTRi = "ListQueueTagsResult";
const _M = "Message";
const _MA = "MessageAttributes";
const _MAN = "MessageAttributeNames";
const _MANe = "MessageAttributeName";
const _MAV = "MessageAttributeValue";
const _MAe = "MessageAttribute";
const _MB = "MessageBody";
const _MBAM = "MessageBodyAttributeMap";
const _MBSAM = "MessageBodySystemAttributeMap";
const _MDI = "MessageDeduplicationId";
const _MDOB = "MD5OfBody";
const _MDOMA = "MD5OfMessageAttributes";
const _MDOMB = "MD5OfMessageBody";
const _MDOMSA = "MD5OfMessageSystemAttributes";
const _MGI = "MessageGroupId";
const _MI = "MessageId";
const _ML = "MessageList";
const _MNI = "MessageNotInflight";
const _MNOM = "MaxNumberOfMessages";
const _MNOMPS = "MaxNumberOfMessagesPerSecond";
const _MR = "MaxResults";
const _MSA = "MessageSystemAttributes";
const _MSAM = "MessageSystemAttributeMap";
const _MSAN = "MessageSystemAttributeNames";
const _MSAV = "MessageSystemAttributeValue";
const _MSAe = "MessageSystemAttribute";
const _Me = "Messages";
const _N = "Name";
const _NT = "NextToken";
const _OL = "OverLimit";
const _PQ = "PurgeQueue";
const _PQIP = "PurgeQueueInProgress";
const _PQR = "PurgeQueueRequest";
const _QAM = "QueueAttributeMap";
const _QDNE = "QueueDoesNotExist";
const _QDR = "QueueDeletedRecently";
const _QN = "QueueName";
const _QNE = "QueueNameExists";
const _QNP = "QueueNamePrefix";
const _QOAWSAI = "QueueOwnerAWSAccountId";
const _QU = "QueueUrl";
const _QUu = "QueueUrls";
const _R = "Results";
const _RH = "ReceiptHandle";
const _RHII = "ReceiptHandleIsInvalid";
const _RM = "ReceiveMessage";
const _RMR = "ReceiveMessageRequest";
const _RMRe = "ReceiveMessageResult";
const _RNFE = "ResourceNotFoundException";
const _RP = "RemovePermission";
const _RPR = "RemovePermissionRequest";
const _RRAI = "ReceiveRequestAttemptId";
const _RT = "RequestThrottled";
const _S = "Successful";
const _SA = "SourceArn";
const _SF = "SenderFault";
const _SL = "StringList";
const _SLV = "StringListValues";
const _SLVt = "StringListValue";
const _SM = "SendMessage";
const _SMB = "SendMessageBatch";
const _SMBR = "SendMessageBatchRequest";
const _SMBRE = "SendMessageBatchRequestEntry";
const _SMBREL = "SendMessageBatchRequestEntryList";
const _SMBRELe = "SendMessageBatchResultEntryList";
const _SMBREe = "SendMessageBatchResultEntry";
const _SMBRe = "SendMessageBatchResult";
const _SMMT = "StartMessageMoveTask";
const _SMMTR = "StartMessageMoveTaskRequest";
const _SMMTRt = "StartMessageMoveTaskResult";
const _SMR = "SendMessageRequest";
const _SMRe = "SendMessageResult";
const _SN = "SequenceNumber";
const _SQA = "SetQueueAttributes";
const _SQAR = "SetQueueAttributesRequest";
const _ST = "StartedTimestamp";
const _SV = "StringValue";
const _St = "Status";
const _T = "Tag";
const _TH = "TaskHandle";
const _TK = "TagKeys";
const _TKa = "TagKey";
const _TM = "TagMap";
const _TMEIBR = "TooManyEntriesInBatchRequest";
const _TQ = "TagQueue";
const _TQR = "TagQueueRequest";
const _Ta = "Tags";
const _UO = "UnsupportedOperation";
const _UQ = "UntagQueue";
const _UQR = "UntagQueueRequest";
const _V = "Value";
const _VT = "VisibilityTimeout";
const _WTS = "WaitTimeSeconds";
const _aQE = "awsQueryError";
const _c = "client";
const _e = "error";
const _hE = "httpError";
const _m = "message";
const _qU = "queueUrls";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.sqs";
const _t = "tags";
const _xF = "xmlFlattened";
const _xN = "xmlName";
const n0 = "com.amazonaws.sqs";
const _s_registry = TypeRegistry.for(_s);
var SQSServiceException$ = [-3, _s, "SQSServiceException", 0, [], []];
_s_registry.registerError(SQSServiceException$, SQSServiceException);
const n0_registry = TypeRegistry.for(n0);
var BatchEntryIdsNotDistinct$ = [-3, n0, _BEIND,
    { [_aQE]: [`AWS.SimpleQueueService.BatchEntryIdsNotDistinct`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(BatchEntryIdsNotDistinct$, BatchEntryIdsNotDistinct);
var BatchRequestTooLong$ = [-3, n0, _BRTL,
    { [_aQE]: [`AWS.SimpleQueueService.BatchRequestTooLong`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(BatchRequestTooLong$, BatchRequestTooLong);
var EmptyBatchRequest$ = [-3, n0, _EBR,
    { [_aQE]: [`AWS.SimpleQueueService.EmptyBatchRequest`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(EmptyBatchRequest$, EmptyBatchRequest);
var InvalidAddress$ = [-3, n0, _IA,
    { [_aQE]: [`InvalidAddress`, 404], [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
n0_registry.registerError(InvalidAddress$, InvalidAddress);
var InvalidAttributeName$ = [-3, n0, _IAN,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(InvalidAttributeName$, InvalidAttributeName);
var InvalidAttributeValue$ = [-3, n0, _IAV,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(InvalidAttributeValue$, InvalidAttributeValue);
var InvalidBatchEntryId$ = [-3, n0, _IBEI,
    { [_aQE]: [`AWS.SimpleQueueService.InvalidBatchEntryId`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(InvalidBatchEntryId$, InvalidBatchEntryId);
var InvalidIdFormat$ = [-3, n0, _IIF,
    { [_e]: _c },
    [],
    []
];
n0_registry.registerError(InvalidIdFormat$, InvalidIdFormat);
var InvalidMessageContents$ = [-3, n0, _IMC,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(InvalidMessageContents$, InvalidMessageContents);
var InvalidSecurity$ = [-3, n0, _IS,
    { [_aQE]: [`InvalidSecurity`, 403], [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
n0_registry.registerError(InvalidSecurity$, InvalidSecurity);
var KmsAccessDenied$ = [-3, n0, _KAD,
    { [_aQE]: [`KMS.AccessDeniedException`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(KmsAccessDenied$, KmsAccessDenied);
var KmsDisabled$ = [-3, n0, _KD,
    { [_aQE]: [`KMS.DisabledException`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(KmsDisabled$, KmsDisabled);
var KmsInvalidKeyUsage$ = [-3, n0, _KIKU,
    { [_aQE]: [`KMS.InvalidKeyUsageException`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(KmsInvalidKeyUsage$, KmsInvalidKeyUsage);
var KmsInvalidState$ = [-3, n0, _KIS,
    { [_aQE]: [`KMS.InvalidStateException`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(KmsInvalidState$, KmsInvalidState);
var KmsNotFound$ = [-3, n0, _KNF,
    { [_aQE]: [`KMS.NotFoundException`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(KmsNotFound$, KmsNotFound);
var KmsOptInRequired$ = [-3, n0, _KOIR,
    { [_aQE]: [`KMS.OptInRequired`, 403], [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
n0_registry.registerError(KmsOptInRequired$, KmsOptInRequired);
var KmsThrottled$ = [-3, n0, _KT,
    { [_aQE]: [`KMS.ThrottlingException`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(KmsThrottled$, KmsThrottled);
var MessageNotInflight$ = [-3, n0, _MNI,
    { [_aQE]: [`AWS.SimpleQueueService.MessageNotInflight`, 400], [_e]: _c, [_hE]: 400 },
    [],
    []
];
n0_registry.registerError(MessageNotInflight$, MessageNotInflight);
var OverLimit$ = [-3, n0, _OL,
    { [_aQE]: [`OverLimit`, 403], [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
n0_registry.registerError(OverLimit$, OverLimit);
var PurgeQueueInProgress$ = [-3, n0, _PQIP,
    { [_aQE]: [`AWS.SimpleQueueService.PurgeQueueInProgress`, 403], [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
n0_registry.registerError(PurgeQueueInProgress$, PurgeQueueInProgress);
var QueueDeletedRecently$ = [-3, n0, _QDR,
    { [_aQE]: [`AWS.SimpleQueueService.QueueDeletedRecently`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(QueueDeletedRecently$, QueueDeletedRecently);
var QueueDoesNotExist$ = [-3, n0, _QDNE,
    { [_aQE]: [`AWS.SimpleQueueService.NonExistentQueue`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(QueueDoesNotExist$, QueueDoesNotExist);
var QueueNameExists$ = [-3, n0, _QNE,
    { [_aQE]: [`QueueAlreadyExists`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(QueueNameExists$, QueueNameExists);
var ReceiptHandleIsInvalid$ = [-3, n0, _RHII,
    { [_aQE]: [`ReceiptHandleIsInvalid`, 404], [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
n0_registry.registerError(ReceiptHandleIsInvalid$, ReceiptHandleIsInvalid);
var RequestThrottled$ = [-3, n0, _RT,
    { [_aQE]: [`RequestThrottled`, 403], [_e]: _c, [_hE]: 403 },
    [_m],
    [0]
];
n0_registry.registerError(RequestThrottled$, RequestThrottled);
var ResourceNotFoundException$ = [-3, n0, _RNFE,
    { [_aQE]: [`ResourceNotFoundException`, 404], [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
n0_registry.registerError(ResourceNotFoundException$, ResourceNotFoundException);
var TooManyEntriesInBatchRequest$ = [-3, n0, _TMEIBR,
    { [_aQE]: [`AWS.SimpleQueueService.TooManyEntriesInBatchRequest`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(TooManyEntriesInBatchRequest$, TooManyEntriesInBatchRequest);
var UnsupportedOperation$ = [-3, n0, _UO,
    { [_aQE]: [`AWS.SimpleQueueService.UnsupportedOperation`, 400], [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(UnsupportedOperation$, UnsupportedOperation);
const errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
var AddPermissionRequest$ = [3, n0, _APR,
    0,
    [_QU, _L, _AWSAI, _A],
    [0, 0, [64 | 0, { [_xF]: 1, [_xN]: _AWSAIc }], [64 | 0, { [_xF]: 1, [_xN]: _AN }]], 4
];
var BatchResultErrorEntry$ = [3, n0, _BREE,
    0,
    [_I, _SF, _C, _M],
    [0, 2, 0, 0], 3
];
var CancelMessageMoveTaskRequest$ = [3, n0, _CMMTR,
    0,
    [_TH],
    [0], 1
];
var CancelMessageMoveTaskResult$ = [3, n0, _CMMTRa,
    0,
    [_ANOMM],
    [1]
];
var ChangeMessageVisibilityBatchRequest$ = [3, n0, _CMVBR,
    0,
    [_QU, _E],
    [0, [() => ChangeMessageVisibilityBatchRequestEntryList, { [_xF]: 1, [_xN]: _CMVBRE }]], 2
];
var ChangeMessageVisibilityBatchRequestEntry$ = [3, n0, _CMVBRE,
    0,
    [_I, _RH, _VT],
    [0, 0, 1], 2
];
var ChangeMessageVisibilityBatchResult$ = [3, n0, _CMVBRh,
    0,
    [_S, _F],
    [[() => ChangeMessageVisibilityBatchResultEntryList, { [_xF]: 1, [_xN]: _CMVBREh }], [() => BatchResultErrorEntryList, { [_xF]: 1, [_xN]: _BREE }]], 2
];
var ChangeMessageVisibilityBatchResultEntry$ = [3, n0, _CMVBREh,
    0,
    [_I],
    [0], 1
];
var ChangeMessageVisibilityRequest$ = [3, n0, _CMVR,
    0,
    [_QU, _RH, _VT],
    [0, 0, 1], 3
];
var CreateQueueRequest$ = [3, n0, _CQR,
    0,
    [_QN, _At, _t],
    [0, [() => QueueAttributeMap, { [_xF]: 1, [_xN]: _Att }], [() => TagMap, { [_xF]: 1, [_xN]: _T }]], 1
];
var CreateQueueResult$ = [3, n0, _CQRr,
    0,
    [_QU],
    [0]
];
var DeleteMessageBatchRequest$ = [3, n0, _DMBR,
    0,
    [_QU, _E],
    [0, [() => DeleteMessageBatchRequestEntryList, { [_xF]: 1, [_xN]: _DMBRE }]], 2
];
var DeleteMessageBatchRequestEntry$ = [3, n0, _DMBRE,
    0,
    [_I, _RH],
    [0, 0], 2
];
var DeleteMessageBatchResult$ = [3, n0, _DMBRe,
    0,
    [_S, _F],
    [[() => DeleteMessageBatchResultEntryList, { [_xF]: 1, [_xN]: _DMBREe }], [() => BatchResultErrorEntryList, { [_xF]: 1, [_xN]: _BREE }]], 2
];
var DeleteMessageBatchResultEntry$ = [3, n0, _DMBREe,
    0,
    [_I],
    [0], 1
];
var DeleteMessageRequest$ = [3, n0, _DMR,
    0,
    [_QU, _RH],
    [0, 0], 2
];
var DeleteQueueRequest$ = [3, n0, _DQR,
    0,
    [_QU],
    [0], 1
];
var GetQueueAttributesRequest$ = [3, n0, _GQAR,
    0,
    [_QU, _ANt],
    [0, [64 | 0, { [_xF]: 1, [_xN]: _ANtt }]], 1
];
var GetQueueAttributesResult$ = [3, n0, _GQARe,
    0,
    [_At],
    [[() => QueueAttributeMap, { [_xF]: 1, [_xN]: _Att }]]
];
var GetQueueUrlRequest$ = [3, n0, _GQUR,
    0,
    [_QN, _QOAWSAI],
    [0, 0], 1
];
var GetQueueUrlResult$ = [3, n0, _GQURe,
    0,
    [_QU],
    [0]
];
var ListDeadLetterSourceQueuesRequest$ = [3, n0, _LDLSQR,
    0,
    [_QU, _NT, _MR],
    [0, 0, 1], 1
];
var ListDeadLetterSourceQueuesResult$ = [3, n0, _LDLSQRi,
    0,
    [_qU, _NT],
    [[64 | 0, { [_xF]: 1, [_xN]: _QU }], 0], 1
];
var ListMessageMoveTasksRequest$ = [3, n0, _LMMTR,
    0,
    [_SA, _MR],
    [0, 1], 1
];
var ListMessageMoveTasksResult$ = [3, n0, _LMMTRi,
    { [_xN]: _LMMTRi },
    [_R],
    [[() => ListMessageMoveTasksResultEntryList, { [_xF]: 1, [_xN]: _LMMTRE }]]
];
var ListMessageMoveTasksResultEntry$ = [3, n0, _LMMTRE,
    0,
    [_TH, _St, _SA, _DA, _MNOMPS, _ANOMM, _ANOMTM, _FR, _ST],
    [0, 0, 0, 0, 1, 1, 1, 0, 1]
];
var ListQueuesRequest$ = [3, n0, _LQR,
    0,
    [_QNP, _NT, _MR],
    [0, 0, 1]
];
var ListQueuesResult$ = [3, n0, _LQRi,
    0,
    [_QUu, _NT],
    [[64 | 0, { [_xF]: 1, [_xN]: _QU }], 0]
];
var ListQueueTagsRequest$ = [3, n0, _LQTR,
    0,
    [_QU],
    [0], 1
];
var ListQueueTagsResult$ = [3, n0, _LQTRi,
    0,
    [_Ta],
    [[() => TagMap, { [_xF]: 1, [_xN]: _T }]]
];
var Message$ = [3, n0, _M,
    0,
    [_MI, _RH, _MDOB, _B, _At, _MDOMA, _MA],
    [0, 0, 0, 0, [() => MessageSystemAttributeMap, { [_xF]: 1, [_xN]: _Att }], 0, [() => MessageBodyAttributeMap, { [_xF]: 1, [_xN]: _MAe }]]
];
var MessageAttributeValue$ = [3, n0, _MAV,
    0,
    [_DT, _SV, _BV, _SLV, _BLV],
    [0, 0, 21, [() => StringList, { [_xF]: 1, [_xN]: _SLVt }], [() => BinaryList, { [_xF]: 1, [_xN]: _BLVi }]], 1
];
var MessageSystemAttributeValue$ = [3, n0, _MSAV,
    0,
    [_DT, _SV, _BV, _SLV, _BLV],
    [0, 0, 21, [() => StringList, { [_xF]: 1, [_xN]: _SLVt }], [() => BinaryList, { [_xF]: 1, [_xN]: _BLVi }]], 1
];
var PurgeQueueRequest$ = [3, n0, _PQR,
    0,
    [_QU],
    [0], 1
];
var ReceiveMessageRequest$ = [3, n0, _RMR,
    0,
    [_QU, _ANt, _MSAN, _MAN, _MNOM, _VT, _WTS, _RRAI],
    [0, [64 | 0, { [_xF]: 1, [_xN]: _ANtt }], [64 | 0, { [_xF]: 1, [_xN]: _ANtt }], [64 | 0, { [_xF]: 1, [_xN]: _MANe }], 1, 1, 1, 0], 1
];
var ReceiveMessageResult$ = [3, n0, _RMRe,
    0,
    [_Me],
    [[() => MessageList, { [_xF]: 1, [_xN]: _M }]]
];
var RemovePermissionRequest$ = [3, n0, _RPR,
    0,
    [_QU, _L],
    [0, 0], 2
];
var SendMessageBatchRequest$ = [3, n0, _SMBR,
    0,
    [_QU, _E],
    [0, [() => SendMessageBatchRequestEntryList, { [_xF]: 1, [_xN]: _SMBRE }]], 2
];
var SendMessageBatchRequestEntry$ = [3, n0, _SMBRE,
    0,
    [_I, _MB, _DS, _MA, _MSA, _MDI, _MGI],
    [0, 0, 1, [() => MessageBodyAttributeMap, { [_xF]: 1, [_xN]: _MAe }], [() => MessageBodySystemAttributeMap, { [_xF]: 1, [_xN]: _MSAe }], 0, 0], 2
];
var SendMessageBatchResult$ = [3, n0, _SMBRe,
    0,
    [_S, _F],
    [[() => SendMessageBatchResultEntryList, { [_xF]: 1, [_xN]: _SMBREe }], [() => BatchResultErrorEntryList, { [_xF]: 1, [_xN]: _BREE }]], 2
];
var SendMessageBatchResultEntry$ = [3, n0, _SMBREe,
    0,
    [_I, _MI, _MDOMB, _MDOMA, _MDOMSA, _SN],
    [0, 0, 0, 0, 0, 0], 3
];
var SendMessageRequest$ = [3, n0, _SMR,
    0,
    [_QU, _MB, _DS, _MA, _MSA, _MDI, _MGI],
    [0, 0, 1, [() => MessageBodyAttributeMap, { [_xF]: 1, [_xN]: _MAe }], [() => MessageBodySystemAttributeMap, { [_xF]: 1, [_xN]: _MSAe }], 0, 0], 2
];
var SendMessageResult$ = [3, n0, _SMRe,
    0,
    [_MDOMB, _MDOMA, _MDOMSA, _MI, _SN],
    [0, 0, 0, 0, 0]
];
var SetQueueAttributesRequest$ = [3, n0, _SQAR,
    0,
    [_QU, _At],
    [0, [() => QueueAttributeMap, { [_xF]: 1, [_xN]: _Att }]], 2
];
var StartMessageMoveTaskRequest$ = [3, n0, _SMMTR,
    0,
    [_SA, _DA, _MNOMPS],
    [0, 0, 1], 1
];
var StartMessageMoveTaskResult$ = [3, n0, _SMMTRt,
    0,
    [_TH],
    [0]
];
var TagQueueRequest$ = [3, n0, _TQR,
    0,
    [_QU, _Ta],
    [0, [() => TagMap, { [_xF]: 1, [_xN]: _T }]], 2
];
var UntagQueueRequest$ = [3, n0, _UQR,
    0,
    [_QU, _TK],
    [0, [64 | 0, { [_xF]: 1, [_xN]: _TKa }]], 2
];
var __Unit = "unit";
var BatchResultErrorEntryList = [1, n0, _BREEL,
    0, () => BatchResultErrorEntry$
];
var BinaryList = [1, n0, _BL,
    0, [21,
        { [_xN]: _BLVi }]
];
var ChangeMessageVisibilityBatchRequestEntryList = [1, n0, _CMVBREL,
    0, () => ChangeMessageVisibilityBatchRequestEntry$
];
var ChangeMessageVisibilityBatchResultEntryList = [1, n0, _CMVBRELh,
    0, () => ChangeMessageVisibilityBatchResultEntry$
];
var DeleteMessageBatchRequestEntryList = [1, n0, _DMBREL,
    0, () => DeleteMessageBatchRequestEntry$
];
var DeleteMessageBatchResultEntryList = [1, n0, _DMBRELe,
    0, () => DeleteMessageBatchResultEntry$
];
var ListMessageMoveTasksResultEntryList = [1, n0, _LMMTREL,
    0, () => ListMessageMoveTasksResultEntry$
];
var MessageList = [1, n0, _ML,
    0, [() => Message$,
        0]
];
var SendMessageBatchRequestEntryList = [1, n0, _SMBREL,
    0, [() => SendMessageBatchRequestEntry$,
        0]
];
var SendMessageBatchResultEntryList = [1, n0, _SMBRELe,
    0, () => SendMessageBatchResultEntry$
];
var StringList = [1, n0, _SL,
    0, [0,
        { [_xN]: _SLVt }]
];
var MessageBodyAttributeMap = [2, n0, _MBAM,
    0, [0,
        { [_xN]: _N }],
    [() => MessageAttributeValue$,
        { [_xN]: _V }]
];
var MessageBodySystemAttributeMap = [2, n0, _MBSAM,
    0, [0,
        { [_xN]: _N }],
    [() => MessageSystemAttributeValue$,
        { [_xN]: _V }]
];
var MessageSystemAttributeMap = [2, n0, _MSAM,
    0, [0,
        { [_xN]: _N }],
    [0,
        { [_xN]: _V }]
];
var QueueAttributeMap = [2, n0, _QAM,
    0, [0,
        { [_xN]: _N }],
    [0,
        { [_xN]: _V }]
];
var TagMap = [2, n0, _TM,
    0, [0,
        { [_xN]: _K }],
    [0,
        { [_xN]: _V }]
];
var AddPermission$ = [9, n0, _AP,
    0, () => AddPermissionRequest$, () => __Unit
];
var CancelMessageMoveTask$ = [9, n0, _CMMT,
    0, () => CancelMessageMoveTaskRequest$, () => CancelMessageMoveTaskResult$
];
var ChangeMessageVisibility$ = [9, n0, _CMV,
    0, () => ChangeMessageVisibilityRequest$, () => __Unit
];
var ChangeMessageVisibilityBatch$ = [9, n0, _CMVB,
    0, () => ChangeMessageVisibilityBatchRequest$, () => ChangeMessageVisibilityBatchResult$
];
var CreateQueue$ = [9, n0, _CQ,
    0, () => CreateQueueRequest$, () => CreateQueueResult$
];
var DeleteMessage$ = [9, n0, _DM,
    0, () => DeleteMessageRequest$, () => __Unit
];
var DeleteMessageBatch$ = [9, n0, _DMB,
    0, () => DeleteMessageBatchRequest$, () => DeleteMessageBatchResult$
];
var DeleteQueue$ = [9, n0, _DQ,
    0, () => DeleteQueueRequest$, () => __Unit
];
var GetQueueAttributes$ = [9, n0, _GQA,
    0, () => GetQueueAttributesRequest$, () => GetQueueAttributesResult$
];
var GetQueueUrl$ = [9, n0, _GQU,
    0, () => GetQueueUrlRequest$, () => GetQueueUrlResult$
];
var ListDeadLetterSourceQueues$ = [9, n0, _LDLSQ,
    0, () => ListDeadLetterSourceQueuesRequest$, () => ListDeadLetterSourceQueuesResult$
];
var ListMessageMoveTasks$ = [9, n0, _LMMT,
    0, () => ListMessageMoveTasksRequest$, () => ListMessageMoveTasksResult$
];
var ListQueues$ = [9, n0, _LQ,
    0, () => ListQueuesRequest$, () => ListQueuesResult$
];
var ListQueueTags$ = [9, n0, _LQT,
    0, () => ListQueueTagsRequest$, () => ListQueueTagsResult$
];
var PurgeQueue$ = [9, n0, _PQ,
    0, () => PurgeQueueRequest$, () => __Unit
];
var ReceiveMessage$ = [9, n0, _RM,
    0, () => ReceiveMessageRequest$, () => ReceiveMessageResult$
];
var RemovePermission$ = [9, n0, _RP,
    0, () => RemovePermissionRequest$, () => __Unit
];
var SendMessage$ = [9, n0, _SM,
    0, () => SendMessageRequest$, () => SendMessageResult$
];
var SendMessageBatch$ = [9, n0, _SMB,
    0, () => SendMessageBatchRequest$, () => SendMessageBatchResult$
];
var SetQueueAttributes$ = [9, n0, _SQA,
    0, () => SetQueueAttributesRequest$, () => __Unit
];
var StartMessageMoveTask$ = [9, n0, _SMMT,
    0, () => StartMessageMoveTaskRequest$, () => StartMessageMoveTaskResult$
];
var TagQueue$ = [9, n0, _TQ,
    0, () => TagQueueRequest$, () => __Unit
];
var UntagQueue$ = [9, n0, _UQ,
    0, () => UntagQueueRequest$, () => __Unit
];

const getRuntimeConfig$1 = (config) => {
    return {
        apiVersion: "2012-11-05",
        base64Decoder: config?.base64Decoder ?? fromBase64,
        base64Encoder: config?.base64Encoder ?? toBase64,
        disableHostPrefix: config?.disableHostPrefix ?? false,
        endpointProvider: config?.endpointProvider ?? defaultEndpointResolver,
        extensions: config?.extensions ?? [],
        httpAuthSchemeProvider: config?.httpAuthSchemeProvider ?? defaultSQSHttpAuthSchemeProvider,
        httpAuthSchemes: config?.httpAuthSchemes ?? [
            {
                schemeId: "aws.auth#sigv4",
                identityProvider: (ipc) => ipc.getIdentityProvider("aws.auth#sigv4"),
                signer: new AwsSdkSigV4Signer(),
            },
        ],
        logger: config?.logger ?? new NoOpLogger(),
        md5: config?.md5 ?? Md5,
        protocol: config?.protocol ?? AwsJson1_0Protocol,
        protocolSettings: config?.protocolSettings ?? {
            defaultNamespace: "com.amazonaws.sqs",
            errorTypeRegistries,
            version: "2012-11-05",
            serviceTarget: "AmazonSQS",
            awsQueryCompatible: true,
        },
        serviceId: config?.serviceId ?? "SQS",
        sha256: config?.sha256 ?? Sha256,
        urlParser: config?.urlParser ?? parseUrl,
        utf8Decoder: config?.utf8Decoder ?? fromUtf8,
        utf8Encoder: config?.utf8Encoder ?? toUtf8,
    };
};

const getRuntimeConfig = (config) => {
    emitWarningIfUnsupportedVersion(process.version);
    const defaultsMode = resolveDefaultsModeConfig(config);
    const defaultConfigProvider = () => defaultsMode().then(loadConfigsForDefaultMode);
    const clientSharedValues = getRuntimeConfig$1(config);
    emitWarningIfUnsupportedVersion$1(process.version);
    const loaderConfig = {
        profile: config?.profile,
        logger: clientSharedValues.logger,
    };
    return {
        ...clientSharedValues,
        ...config,
        runtime: "node",
        defaultsMode,
        authSchemePreference: config?.authSchemePreference ?? loadConfig(NODE_AUTH_SCHEME_PREFERENCE_OPTIONS, loaderConfig),
        bodyLengthChecker: config?.bodyLengthChecker ?? calculateBodyLength,
        credentialDefaultProvider: config?.credentialDefaultProvider ?? defaultProvider,
        defaultUserAgentProvider: config?.defaultUserAgentProvider ?? createDefaultUserAgentProvider({ serviceId: clientSharedValues.serviceId, clientVersion: packageInfo.version }),
        maxAttempts: config?.maxAttempts ?? loadConfig(NODE_MAX_ATTEMPT_CONFIG_OPTIONS, config),
        region: config?.region ?? loadConfig(NODE_REGION_CONFIG_OPTIONS, { ...NODE_REGION_CONFIG_FILE_OPTIONS, ...loaderConfig }),
        requestHandler: NodeHttpHandler.create(config?.requestHandler ?? defaultConfigProvider),
        retryMode: config?.retryMode ??
            loadConfig({
                ...NODE_RETRY_MODE_CONFIG_OPTIONS,
                default: async () => (await defaultConfigProvider()).retryMode || DEFAULT_RETRY_MODE,
            }, config),
        streamCollector: config?.streamCollector ?? streamCollector,
        useDualstackEndpoint: config?.useDualstackEndpoint ?? loadConfig(NODE_USE_DUALSTACK_ENDPOINT_CONFIG_OPTIONS, loaderConfig),
        useFipsEndpoint: config?.useFipsEndpoint ?? loadConfig(NODE_USE_FIPS_ENDPOINT_CONFIG_OPTIONS, loaderConfig),
        userAgentAppId: config?.userAgentAppId ?? loadConfig(NODE_APP_ID_CONFIG_OPTIONS, loaderConfig),
    };
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(getAwsRegionExtensionConfiguration(runtimeConfig), getDefaultExtensionConfiguration(runtimeConfig), getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, resolveAwsRegionExtensionConfiguration(extensionConfiguration), resolveDefaultRuntimeConfig(extensionConfiguration), resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class SQSClient extends Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = resolveUserAgentConfig(_config_1);
        const _config_3 = resolveRetryConfig(_config_2);
        const _config_4 = resolveRegionConfig(_config_3);
        const _config_5 = resolveHostHeaderConfig(_config_4);
        const _config_6 = resolveEndpointConfig(_config_5);
        const _config_7 = resolveQueueUrlConfig(_config_6);
        const _config_8 = resolveHttpAuthSchemeConfig(_config_7);
        const _config_9 = resolveRuntimeExtensions(_config_8, configuration?.extensions || []);
        this.config = _config_9;
        this.middlewareStack.use(getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(getUserAgentPlugin(this.config));
        this.middlewareStack.use(getRetryPlugin(this.config));
        this.middlewareStack.use(getContentLengthPlugin(this.config));
        this.middlewareStack.use(getHostHeaderPlugin(this.config));
        this.middlewareStack.use(getLoggerPlugin(this.config));
        this.middlewareStack.use(getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(getQueueUrlPlugin(this.config));
        this.middlewareStack.use(getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: defaultSQSHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

const command = makeBuilder(commonParams, "AmazonSQS", "SQSClient", getEndpointPlugin);
const _ep0 = {};
const _mw0 = (Command, cs, config, o) => [];
const _mw1 = (Command, cs, config, o) => [
    getReceiveMessagePlugin(config),
    getLongPollPlugin(config),
];
const _mw2 = (Command, cs, config, o) => [
    getSendMessageBatchPlugin(config),
];
const _mw3 = (Command, cs, config, o) => [
    getSendMessagePlugin(config),
];

class AddPermissionCommand extends command(_ep0, _mw0, "AddPermission", AddPermission$) {
}

class CancelMessageMoveTaskCommand extends command(_ep0, _mw0, "CancelMessageMoveTask", CancelMessageMoveTask$) {
}

class ChangeMessageVisibilityBatchCommand extends command(_ep0, _mw0, "ChangeMessageVisibilityBatch", ChangeMessageVisibilityBatch$) {
}

class ChangeMessageVisibilityCommand extends command(_ep0, _mw0, "ChangeMessageVisibility", ChangeMessageVisibility$) {
}

class CreateQueueCommand extends command(_ep0, _mw0, "CreateQueue", CreateQueue$) {
}

class DeleteMessageBatchCommand extends command(_ep0, _mw0, "DeleteMessageBatch", DeleteMessageBatch$) {
}

class DeleteMessageCommand extends command(_ep0, _mw0, "DeleteMessage", DeleteMessage$) {
}

class DeleteQueueCommand extends command(_ep0, _mw0, "DeleteQueue", DeleteQueue$) {
}

class GetQueueAttributesCommand extends command(_ep0, _mw0, "GetQueueAttributes", GetQueueAttributes$) {
}

class GetQueueUrlCommand extends command(_ep0, _mw0, "GetQueueUrl", GetQueueUrl$) {
}

class ListDeadLetterSourceQueuesCommand extends command(_ep0, _mw0, "ListDeadLetterSourceQueues", ListDeadLetterSourceQueues$) {
}

class ListMessageMoveTasksCommand extends command(_ep0, _mw0, "ListMessageMoveTasks", ListMessageMoveTasks$) {
}

class ListQueuesCommand extends command(_ep0, _mw0, "ListQueues", ListQueues$) {
}

class ListQueueTagsCommand extends command(_ep0, _mw0, "ListQueueTags", ListQueueTags$) {
}

class PurgeQueueCommand extends command(_ep0, _mw0, "PurgeQueue", PurgeQueue$) {
}

class ReceiveMessageCommand extends command(_ep0, _mw1, "ReceiveMessage", ReceiveMessage$) {
}

class RemovePermissionCommand extends command(_ep0, _mw0, "RemovePermission", RemovePermission$) {
}

class SendMessageBatchCommand extends command(_ep0, _mw2, "SendMessageBatch", SendMessageBatch$) {
}

class SendMessageCommand extends command(_ep0, _mw3, "SendMessage", SendMessage$) {
}

class SetQueueAttributesCommand extends command(_ep0, _mw0, "SetQueueAttributes", SetQueueAttributes$) {
}

class StartMessageMoveTaskCommand extends command(_ep0, _mw0, "StartMessageMoveTask", StartMessageMoveTask$) {
}

class TagQueueCommand extends command(_ep0, _mw0, "TagQueue", TagQueue$) {
}

class UntagQueueCommand extends command(_ep0, _mw0, "UntagQueue", UntagQueue$) {
}

const paginateListDeadLetterSourceQueues = createPaginator(SQSClient, ListDeadLetterSourceQueuesCommand, "NextToken", "NextToken", "MaxResults");

const paginateListQueues = createPaginator(SQSClient, ListQueuesCommand, "NextToken", "NextToken", "MaxResults");

const commands = {
    AddPermissionCommand,
    CancelMessageMoveTaskCommand,
    ChangeMessageVisibilityCommand,
    ChangeMessageVisibilityBatchCommand,
    CreateQueueCommand,
    DeleteMessageCommand,
    DeleteMessageBatchCommand,
    DeleteQueueCommand,
    GetQueueAttributesCommand,
    GetQueueUrlCommand,
    ListDeadLetterSourceQueuesCommand,
    ListMessageMoveTasksCommand,
    ListQueuesCommand,
    ListQueueTagsCommand,
    PurgeQueueCommand,
    ReceiveMessageCommand,
    RemovePermissionCommand,
    SendMessageCommand,
    SendMessageBatchCommand,
    SetQueueAttributesCommand,
    StartMessageMoveTaskCommand,
    TagQueueCommand,
    UntagQueueCommand,
};
const paginators = {
    paginateListDeadLetterSourceQueues,
    paginateListQueues,
};
class SQS extends SQSClient {
}
createAggregatedClient(commands, SQS, { paginators });

const QueueAttributeName = {
    All: "All",
    ApproximateNumberOfMessages: "ApproximateNumberOfMessages",
    ApproximateNumberOfMessagesDelayed: "ApproximateNumberOfMessagesDelayed",
    ApproximateNumberOfMessagesNotVisible: "ApproximateNumberOfMessagesNotVisible",
    ContentBasedDeduplication: "ContentBasedDeduplication",
    CreatedTimestamp: "CreatedTimestamp",
    DeduplicationScope: "DeduplicationScope",
    DelaySeconds: "DelaySeconds",
    FifoQueue: "FifoQueue",
    FifoThroughputLimit: "FifoThroughputLimit",
    KmsDataKeyReusePeriodSeconds: "KmsDataKeyReusePeriodSeconds",
    KmsMasterKeyId: "KmsMasterKeyId",
    LastModifiedTimestamp: "LastModifiedTimestamp",
    MaximumMessageSize: "MaximumMessageSize",
    MessageRetentionPeriod: "MessageRetentionPeriod",
    Policy: "Policy",
    QueueArn: "QueueArn",
    ReceiveMessageWaitTimeSeconds: "ReceiveMessageWaitTimeSeconds",
    RedriveAllowPolicy: "RedriveAllowPolicy",
    RedrivePolicy: "RedrivePolicy",
    SqsManagedSseEnabled: "SqsManagedSseEnabled",
    VisibilityTimeout: "VisibilityTimeout",
};
const MessageSystemAttributeName = {
    AWSTraceHeader: "AWSTraceHeader",
    All: "All",
    ApproximateFirstReceiveTimestamp: "ApproximateFirstReceiveTimestamp",
    ApproximateReceiveCount: "ApproximateReceiveCount",
    DeadLetterQueueSourceArn: "DeadLetterQueueSourceArn",
    MessageDeduplicationId: "MessageDeduplicationId",
    MessageGroupId: "MessageGroupId",
    SenderId: "SenderId",
    SentTimestamp: "SentTimestamp",
    SequenceNumber: "SequenceNumber",
};
const MessageSystemAttributeNameForSends = {
    AWSTraceHeader: "AWSTraceHeader",
};

exports.AddPermission$ = AddPermission$;
exports.AddPermissionCommand = AddPermissionCommand;
exports.AddPermissionRequest$ = AddPermissionRequest$;
exports.BatchEntryIdsNotDistinct = BatchEntryIdsNotDistinct;
exports.BatchEntryIdsNotDistinct$ = BatchEntryIdsNotDistinct$;
exports.BatchRequestTooLong = BatchRequestTooLong;
exports.BatchRequestTooLong$ = BatchRequestTooLong$;
exports.BatchResultErrorEntry$ = BatchResultErrorEntry$;
exports.CancelMessageMoveTask$ = CancelMessageMoveTask$;
exports.CancelMessageMoveTaskCommand = CancelMessageMoveTaskCommand;
exports.CancelMessageMoveTaskRequest$ = CancelMessageMoveTaskRequest$;
exports.CancelMessageMoveTaskResult$ = CancelMessageMoveTaskResult$;
exports.ChangeMessageVisibility$ = ChangeMessageVisibility$;
exports.ChangeMessageVisibilityBatch$ = ChangeMessageVisibilityBatch$;
exports.ChangeMessageVisibilityBatchCommand = ChangeMessageVisibilityBatchCommand;
exports.ChangeMessageVisibilityBatchRequest$ = ChangeMessageVisibilityBatchRequest$;
exports.ChangeMessageVisibilityBatchRequestEntry$ = ChangeMessageVisibilityBatchRequestEntry$;
exports.ChangeMessageVisibilityBatchResult$ = ChangeMessageVisibilityBatchResult$;
exports.ChangeMessageVisibilityBatchResultEntry$ = ChangeMessageVisibilityBatchResultEntry$;
exports.ChangeMessageVisibilityCommand = ChangeMessageVisibilityCommand;
exports.ChangeMessageVisibilityRequest$ = ChangeMessageVisibilityRequest$;
exports.CreateQueue$ = CreateQueue$;
exports.CreateQueueCommand = CreateQueueCommand;
exports.CreateQueueRequest$ = CreateQueueRequest$;
exports.CreateQueueResult$ = CreateQueueResult$;
exports.DeleteMessage$ = DeleteMessage$;
exports.DeleteMessageBatch$ = DeleteMessageBatch$;
exports.DeleteMessageBatchCommand = DeleteMessageBatchCommand;
exports.DeleteMessageBatchRequest$ = DeleteMessageBatchRequest$;
exports.DeleteMessageBatchRequestEntry$ = DeleteMessageBatchRequestEntry$;
exports.DeleteMessageBatchResult$ = DeleteMessageBatchResult$;
exports.DeleteMessageBatchResultEntry$ = DeleteMessageBatchResultEntry$;
exports.DeleteMessageCommand = DeleteMessageCommand;
exports.DeleteMessageRequest$ = DeleteMessageRequest$;
exports.DeleteQueue$ = DeleteQueue$;
exports.DeleteQueueCommand = DeleteQueueCommand;
exports.DeleteQueueRequest$ = DeleteQueueRequest$;
exports.EmptyBatchRequest = EmptyBatchRequest;
exports.EmptyBatchRequest$ = EmptyBatchRequest$;
exports.GetQueueAttributes$ = GetQueueAttributes$;
exports.GetQueueAttributesCommand = GetQueueAttributesCommand;
exports.GetQueueAttributesRequest$ = GetQueueAttributesRequest$;
exports.GetQueueAttributesResult$ = GetQueueAttributesResult$;
exports.GetQueueUrl$ = GetQueueUrl$;
exports.GetQueueUrlCommand = GetQueueUrlCommand;
exports.GetQueueUrlRequest$ = GetQueueUrlRequest$;
exports.GetQueueUrlResult$ = GetQueueUrlResult$;
exports.InvalidAddress = InvalidAddress;
exports.InvalidAddress$ = InvalidAddress$;
exports.InvalidAttributeName = InvalidAttributeName;
exports.InvalidAttributeName$ = InvalidAttributeName$;
exports.InvalidAttributeValue = InvalidAttributeValue;
exports.InvalidAttributeValue$ = InvalidAttributeValue$;
exports.InvalidBatchEntryId = InvalidBatchEntryId;
exports.InvalidBatchEntryId$ = InvalidBatchEntryId$;
exports.InvalidIdFormat = InvalidIdFormat;
exports.InvalidIdFormat$ = InvalidIdFormat$;
exports.InvalidMessageContents = InvalidMessageContents;
exports.InvalidMessageContents$ = InvalidMessageContents$;
exports.InvalidSecurity = InvalidSecurity;
exports.InvalidSecurity$ = InvalidSecurity$;
exports.KmsAccessDenied = KmsAccessDenied;
exports.KmsAccessDenied$ = KmsAccessDenied$;
exports.KmsDisabled = KmsDisabled;
exports.KmsDisabled$ = KmsDisabled$;
exports.KmsInvalidKeyUsage = KmsInvalidKeyUsage;
exports.KmsInvalidKeyUsage$ = KmsInvalidKeyUsage$;
exports.KmsInvalidState = KmsInvalidState;
exports.KmsInvalidState$ = KmsInvalidState$;
exports.KmsNotFound = KmsNotFound;
exports.KmsNotFound$ = KmsNotFound$;
exports.KmsOptInRequired = KmsOptInRequired;
exports.KmsOptInRequired$ = KmsOptInRequired$;
exports.KmsThrottled = KmsThrottled;
exports.KmsThrottled$ = KmsThrottled$;
exports.ListDeadLetterSourceQueues$ = ListDeadLetterSourceQueues$;
exports.ListDeadLetterSourceQueuesCommand = ListDeadLetterSourceQueuesCommand;
exports.ListDeadLetterSourceQueuesRequest$ = ListDeadLetterSourceQueuesRequest$;
exports.ListDeadLetterSourceQueuesResult$ = ListDeadLetterSourceQueuesResult$;
exports.ListMessageMoveTasks$ = ListMessageMoveTasks$;
exports.ListMessageMoveTasksCommand = ListMessageMoveTasksCommand;
exports.ListMessageMoveTasksRequest$ = ListMessageMoveTasksRequest$;
exports.ListMessageMoveTasksResult$ = ListMessageMoveTasksResult$;
exports.ListMessageMoveTasksResultEntry$ = ListMessageMoveTasksResultEntry$;
exports.ListQueueTags$ = ListQueueTags$;
exports.ListQueueTagsCommand = ListQueueTagsCommand;
exports.ListQueueTagsRequest$ = ListQueueTagsRequest$;
exports.ListQueueTagsResult$ = ListQueueTagsResult$;
exports.ListQueues$ = ListQueues$;
exports.ListQueuesCommand = ListQueuesCommand;
exports.ListQueuesRequest$ = ListQueuesRequest$;
exports.ListQueuesResult$ = ListQueuesResult$;
exports.Message$ = Message$;
exports.MessageAttributeValue$ = MessageAttributeValue$;
exports.MessageNotInflight = MessageNotInflight;
exports.MessageNotInflight$ = MessageNotInflight$;
exports.MessageSystemAttributeName = MessageSystemAttributeName;
exports.MessageSystemAttributeNameForSends = MessageSystemAttributeNameForSends;
exports.MessageSystemAttributeValue$ = MessageSystemAttributeValue$;
exports.OverLimit = OverLimit;
exports.OverLimit$ = OverLimit$;
exports.PurgeQueue$ = PurgeQueue$;
exports.PurgeQueueCommand = PurgeQueueCommand;
exports.PurgeQueueInProgress = PurgeQueueInProgress;
exports.PurgeQueueInProgress$ = PurgeQueueInProgress$;
exports.PurgeQueueRequest$ = PurgeQueueRequest$;
exports.QueueAttributeName = QueueAttributeName;
exports.QueueDeletedRecently = QueueDeletedRecently;
exports.QueueDeletedRecently$ = QueueDeletedRecently$;
exports.QueueDoesNotExist = QueueDoesNotExist;
exports.QueueDoesNotExist$ = QueueDoesNotExist$;
exports.QueueNameExists = QueueNameExists;
exports.QueueNameExists$ = QueueNameExists$;
exports.ReceiptHandleIsInvalid = ReceiptHandleIsInvalid;
exports.ReceiptHandleIsInvalid$ = ReceiptHandleIsInvalid$;
exports.ReceiveMessage$ = ReceiveMessage$;
exports.ReceiveMessageCommand = ReceiveMessageCommand;
exports.ReceiveMessageRequest$ = ReceiveMessageRequest$;
exports.ReceiveMessageResult$ = ReceiveMessageResult$;
exports.RemovePermission$ = RemovePermission$;
exports.RemovePermissionCommand = RemovePermissionCommand;
exports.RemovePermissionRequest$ = RemovePermissionRequest$;
exports.RequestThrottled = RequestThrottled;
exports.RequestThrottled$ = RequestThrottled$;
exports.ResourceNotFoundException = ResourceNotFoundException;
exports.ResourceNotFoundException$ = ResourceNotFoundException$;
exports.SQS = SQS;
exports.SQSClient = SQSClient;
exports.SQSServiceException = SQSServiceException;
exports.SQSServiceException$ = SQSServiceException$;
exports.SendMessage$ = SendMessage$;
exports.SendMessageBatch$ = SendMessageBatch$;
exports.SendMessageBatchCommand = SendMessageBatchCommand;
exports.SendMessageBatchRequest$ = SendMessageBatchRequest$;
exports.SendMessageBatchRequestEntry$ = SendMessageBatchRequestEntry$;
exports.SendMessageBatchResult$ = SendMessageBatchResult$;
exports.SendMessageBatchResultEntry$ = SendMessageBatchResultEntry$;
exports.SendMessageCommand = SendMessageCommand;
exports.SendMessageRequest$ = SendMessageRequest$;
exports.SendMessageResult$ = SendMessageResult$;
exports.SetQueueAttributes$ = SetQueueAttributes$;
exports.SetQueueAttributesCommand = SetQueueAttributesCommand;
exports.SetQueueAttributesRequest$ = SetQueueAttributesRequest$;
exports.StartMessageMoveTask$ = StartMessageMoveTask$;
exports.StartMessageMoveTaskCommand = StartMessageMoveTaskCommand;
exports.StartMessageMoveTaskRequest$ = StartMessageMoveTaskRequest$;
exports.StartMessageMoveTaskResult$ = StartMessageMoveTaskResult$;
exports.TagQueue$ = TagQueue$;
exports.TagQueueCommand = TagQueueCommand;
exports.TagQueueRequest$ = TagQueueRequest$;
exports.TooManyEntriesInBatchRequest = TooManyEntriesInBatchRequest;
exports.TooManyEntriesInBatchRequest$ = TooManyEntriesInBatchRequest$;
exports.UnsupportedOperation = UnsupportedOperation;
exports.UnsupportedOperation$ = UnsupportedOperation$;
exports.UntagQueue$ = UntagQueue$;
exports.UntagQueueCommand = UntagQueueCommand;
exports.UntagQueueRequest$ = UntagQueueRequest$;
exports.errorTypeRegistries = errorTypeRegistries;
exports.paginateListDeadLetterSourceQueues = paginateListDeadLetterSourceQueues;
exports.paginateListQueues = paginateListQueues;
