// @aws-sdk/client-eventbridge/dist-cjs/index.js
const { awsEndpointFunctions, emitWarningIfUnsupportedVersion: emitWarningIfUnsupportedVersion$1, createDefaultUserAgentProvider, NODE_APP_ID_CONFIG_OPTIONS, getAwsRegionExtensionConfiguration, resolveAwsRegionExtensionConfiguration, resolveUserAgentConfig, resolveHostHeaderConfig, getUserAgentPlugin, getHostHeaderPlugin, getLoggerPlugin, getRecursionDetectionPlugin } = require("@aws-sdk/core/client");
const { getHttpAuthSchemeEndpointRuleSetPlugin, DefaultIdentityProviderConfig, getHttpSigningPlugin } = require("@smithy/core");
const { normalizeProvider, getSmithyContext, ServiceException, NoOpLogger, emitWarningIfUnsupportedVersion, loadConfigsForDefaultMode, getDefaultExtensionConfiguration, resolveDefaultRuntimeConfig, Client, makeBuilder, createAggregatedClient } = require("@smithy/core/client");
const { Command: $Command } = require("@smithy/core/client");
exports.$Command = $Command;
exports.__Client = Client;
const { resolveDefaultsModeConfig, loadConfig, NODE_USE_FIPS_ENDPOINT_CONFIG_OPTIONS, NODE_USE_DUALSTACK_ENDPOINT_CONFIG_OPTIONS, NODE_REGION_CONFIG_OPTIONS, NODE_REGION_CONFIG_FILE_OPTIONS, resolveRegionConfig } = require("@smithy/core/config");
const { BinaryDecisionDiagram, EndpointCache, decideEndpoint, customEndpointFunctions, resolveParams, resolveEndpointConfig, getEndpointPlugin } = require("@smithy/core/endpoints");
const { parseUrl, getHttpHandlerExtensionConfiguration, resolveHttpHandlerRuntimeConfig, getContentLengthPlugin } = require("@smithy/core/protocols");
const { DEFAULT_RETRY_MODE, NODE_RETRY_MODE_CONFIG_OPTIONS, NODE_MAX_ATTEMPT_CONFIG_OPTIONS, resolveRetryConfig, getRetryPlugin } = require("@smithy/core/retry");
const { TypeRegistry, getSchemaSerdePlugin } = require("@smithy/core/schema");
const { resolveAwsSdkSigV4Config, resolveAwsSdkSigV4AConfig, AwsSdkSigV4Signer, AwsSdkSigV4ASigner, NODE_SIGV4A_CONFIG_OPTIONS, NODE_AUTH_SCHEME_PREFERENCE_OPTIONS } = require("@aws-sdk/core/httpAuthSchemes");
const { SignatureV4MultiRegion } = require("@aws-sdk/signature-v4-multi-region");
const { defaultProvider } = require("@aws-sdk/credential-provider-node");
const { toUtf8, fromUtf8, toBase64, fromBase64, calculateBodyLength } = require("@smithy/core/serde");
const { streamCollector, NodeHttpHandler } = require("@smithy/node-http-handler");
const { AwsJson1_1Protocol } = require("@aws-sdk/core/protocols");
const { Sha256 } = require("@smithy/core/checksum");

const o = "ref";
const a = -1, b = true, c = "isSet", d = "PartitionResult", e = "booleanEquals", f = "stringEquals", g = "getAttr", h = { [o]: "Endpoint" }, i = { [o]: "EndpointId" }, j = { [o]: d }, k = { [o]: "Region" }, l = {}, m = { "authSchemes": [{ "name": "sigv4a", "signingName": "events", "signingRegionSet": ["*"] }] }, n = [k];
const _data = {
    conditions: [
        [c, n],
        [c, [h]],
        [c, [i]],
        ["aws.partition", n, d],
        [e, [{ [o]: "UseFIPS" }, b]],
        [e, [{ [o]: "UseDualStack" }, b]],
        [f, [{ fn: g, argv: [j, "name"] }, "aws-us-gov"]],
        ["isValidHostLabel", [i, b]],
        [e, [{ fn: g, argv: [j, "supportsFIPS"] }, b]],
        [e, [{ fn: g, argv: [j, "supportsDualStack"] }, b]],
        [f, [k, "us-gov-east-1"]],
        [f, [k, "us-gov-west-1"]]
    ],
    results: [
        [a],
        ["https://events.{Region}.{PartitionResult#dualStackDnsSuffix}", l],
        [h, m],
        ["https://{EndpointId}.endpoint.events.{PartitionResult#dualStackDnsSuffix}", m],
        [a, "DualStack is enabled but this partition does not support DualStack"],
        ["https://{EndpointId}.endpoint.events.{PartitionResult#dnsSuffix}", m],
        [a, "Invalid Configuration: FIPS is not supported with EventBridge multi-region endpoints."],
        [a, "EndpointId must be a valid host label."],
        [a, "Invalid Configuration: FIPS and custom endpoint are not supported"],
        [a, "Invalid Configuration: Dualstack and custom endpoint are not supported"],
        [h, l],
        ["https://events-fips.{Region}.{PartitionResult#dualStackDnsSuffix}", l],
        [a, "FIPS and DualStack are enabled, but this partition does not support one or both"],
        ["https://events.us-gov-east-1.amazonaws.com", l],
        ["https://events.us-gov-west-1.amazonaws.com", l],
        ["https://events-fips.{Region}.{PartitionResult#dnsSuffix}", l],
        [a, "FIPS is enabled but this partition does not support FIPS"],
        ["https://events.{Region}.{PartitionResult#dnsSuffix}", l],
        [a, "Invalid Configuration: Missing Region"]
    ]
};
const root = 2;
const r = 100_000_000;
const nodes = new Int32Array([
    -1, 1, -1,
    0, 4, 3,
    1, 27, r + 18,
    1, 25, 5,
    2, 17, 6,
    3, 7, r + 18,
    4, 10, 8,
    5, 9, r + 17,
    9, r + 1, r + 4,
    5, 14, 11,
    8, 12, r + 16,
    10, r + 13, 13,
    11, r + 14, r + 15,
    6, r + 1, 15,
    8, 16, r + 12,
    9, r + 11, r + 12,
    3, 18, r + 18,
    4, 23, 19,
    5, 21, 20,
    7, r + 5, r + 7,
    7, 22, r + 7,
    9, r + 3, r + 4,
    5, 24, 31,
    6, r + 1, 31,
    2, 26, 27,
    3, 29, 27,
    4, r + 8, 28,
    5, r + 9, r + 10,
    4, 31, 30,
    7, r + 2, r + 7,
    7, r + 6, r + 7,
]);
const bdd = BinaryDecisionDiagram.from(nodes, root, _data.conditions, _data.results);

const cache = new EndpointCache({
    size: 50,
    params: ["Endpoint", "EndpointId", "Region", "UseDualStack", "UseFIPS"],
});
const defaultEndpointResolver = (endpointParams, context = {}) => {
    return cache.get(endpointParams, () => decideEndpoint(bdd, {
        endpointParams: endpointParams,
        logger: context.logger,
    }));
};
customEndpointFunctions.aws = awsEndpointFunctions;

const createEndpointRuleSetHttpAuthSchemeParametersProvider = (defaultHttpAuthSchemeParametersProvider) => async (config, context, input) => {
    if (!input) {
        throw new Error("Could not find `input` for `defaultEndpointRuleSetHttpAuthSchemeParametersProvider`");
    }
    const defaultParameters = await defaultHttpAuthSchemeParametersProvider(config, context, input);
    const instructionsFn = getSmithyContext(context)?.commandInstance?.constructor
        ?.getEndpointParameterInstructions;
    if (!instructionsFn) {
        throw new Error(`getEndpointParameterInstructions() is not defined on '${context.commandName}'`);
    }
    const endpointParameters = await resolveParams(input, { getEndpointParameterInstructions: instructionsFn }, config);
    return Object.assign(defaultParameters, endpointParameters);
};
const _defaultEventBridgeHttpAuthSchemeParametersProvider = async (config, context, input) => {
    return {
        operation: getSmithyContext(context).operation,
        region: await normalizeProvider(config.region)() || (() => {
            throw new Error("expected `region` to be configured for `aws.auth#sigv4`");
        })(),
    };
};
const defaultEventBridgeHttpAuthSchemeParametersProvider = createEndpointRuleSetHttpAuthSchemeParametersProvider(_defaultEventBridgeHttpAuthSchemeParametersProvider);
function createAwsAuthSigv4HttpAuthOption(authParameters) {
    return {
        schemeId: "aws.auth#sigv4",
        signingProperties: {
            name: "events",
            region: authParameters.region,
        },
        propertiesExtractor: (config, context) => ({
            signingProperties: {
                config,
                context,
            },
        }),
    };
}
function createAwsAuthSigv4aHttpAuthOption(authParameters) {
    return {
        schemeId: "aws.auth#sigv4a",
        signingProperties: {
            name: "events",
            region: authParameters.region,
        },
        propertiesExtractor: (config, context) => ({
            signingProperties: {
                config,
                context,
            },
        }),
    };
}
const createEndpointRuleSetHttpAuthSchemeProvider = (defaultEndpointResolver, defaultHttpAuthSchemeResolver, createHttpAuthOptionFunctions) => {
    const endpointRuleSetHttpAuthSchemeProvider = (authParameters) => {
        const endpoint = defaultEndpointResolver(authParameters);
        const authSchemes = endpoint.properties?.authSchemes;
        if (!authSchemes) {
            return defaultHttpAuthSchemeResolver(authParameters);
        }
        const options = [];
        for (const scheme of authSchemes) {
            const { name: resolvedName, properties = {}, ...rest } = scheme;
            const name = resolvedName.toLowerCase();
            if (resolvedName !== name) {
                console.warn(`HttpAuthScheme has been normalized with lowercasing: '${resolvedName}' to '${name}'`);
            }
            let schemeId;
            if (name === "sigv4a") {
                schemeId = "aws.auth#sigv4a";
                const sigv4Present = authSchemes.find((s) => {
                    const name = s.name.toLowerCase();
                    return name !== "sigv4a" && name.startsWith("sigv4");
                });
                if (SignatureV4MultiRegion.sigv4aDependency() === "none" && sigv4Present) {
                    continue;
                }
            }
            else if (name.startsWith("sigv4")) {
                schemeId = "aws.auth#sigv4";
            }
            else {
                throw new Error(`Unknown HttpAuthScheme found in '@smithy.rules#endpointRuleSet': '${name}'`);
            }
            const createOption = createHttpAuthOptionFunctions[schemeId];
            if (!createOption) {
                throw new Error(`Could not find HttpAuthOption create function for '${schemeId}'`);
            }
            const option = createOption(authParameters);
            option.schemeId = schemeId;
            option.signingProperties = { ...(option.signingProperties || {}), ...rest, ...properties };
            options.push(option);
        }
        return options;
    };
    return endpointRuleSetHttpAuthSchemeProvider;
};
const _defaultEventBridgeHttpAuthSchemeProvider = (authParameters) => {
    const options = [];
    switch (authParameters.operation) {
        default: {
            options.push(createAwsAuthSigv4HttpAuthOption(authParameters));
            options.push(createAwsAuthSigv4aHttpAuthOption(authParameters));
        }
    }
    return options;
};
const defaultEventBridgeHttpAuthSchemeProvider = createEndpointRuleSetHttpAuthSchemeProvider(defaultEndpointResolver, _defaultEventBridgeHttpAuthSchemeProvider, {
    "aws.auth#sigv4": createAwsAuthSigv4HttpAuthOption,
    "aws.auth#sigv4a": createAwsAuthSigv4aHttpAuthOption,
});
const resolveHttpAuthSchemeConfig = (config) => {
    const config_0 = resolveAwsSdkSigV4Config(config);
    const config_1 = resolveAwsSdkSigV4AConfig(config_0);
    return Object.assign(config_1, {
        authSchemePreference: normalizeProvider(config.authSchemePreference ?? []),
    });
};

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "events",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

var version = "3.1084.0";
var packageInfo = {
	version: version};

class EventBridgeServiceException extends ServiceException {
    constructor(options) {
        super(options);
        Object.setPrototypeOf(this, EventBridgeServiceException.prototype);
    }
}

class AccessDeniedException extends EventBridgeServiceException {
    name = "AccessDeniedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "AccessDeniedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AccessDeniedException.prototype);
    }
}
class ConcurrentModificationException extends EventBridgeServiceException {
    name = "ConcurrentModificationException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ConcurrentModificationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConcurrentModificationException.prototype);
    }
}
class InternalException extends EventBridgeServiceException {
    name = "InternalException";
    $fault = "server";
    constructor(opts) {
        super({
            name: "InternalException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalException.prototype);
    }
}
class InvalidStateException extends EventBridgeServiceException {
    name = "InvalidStateException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidStateException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidStateException.prototype);
    }
}
class OperationDisabledException extends EventBridgeServiceException {
    name = "OperationDisabledException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "OperationDisabledException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, OperationDisabledException.prototype);
    }
}
class ResourceNotFoundException extends EventBridgeServiceException {
    name = "ResourceNotFoundException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
    }
}
class IllegalStatusException extends EventBridgeServiceException {
    name = "IllegalStatusException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "IllegalStatusException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, IllegalStatusException.prototype);
    }
}
class LimitExceededException extends EventBridgeServiceException {
    name = "LimitExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "LimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, LimitExceededException.prototype);
    }
}
class ResourceAlreadyExistsException extends EventBridgeServiceException {
    name = "ResourceAlreadyExistsException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ResourceAlreadyExistsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceAlreadyExistsException.prototype);
    }
}
class InvalidEventPatternException extends EventBridgeServiceException {
    name = "InvalidEventPatternException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidEventPatternException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidEventPatternException.prototype);
    }
}
class ThrottlingException extends EventBridgeServiceException {
    name = "ThrottlingException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ThrottlingException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ThrottlingException.prototype);
    }
}
class ManagedRuleException extends EventBridgeServiceException {
    name = "ManagedRuleException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ManagedRuleException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ManagedRuleException.prototype);
    }
}
class PolicyLengthExceededException extends EventBridgeServiceException {
    name = "PolicyLengthExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "PolicyLengthExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, PolicyLengthExceededException.prototype);
    }
}

const _A = "Archive";
const _AA = "ArchiveArn";
const _AD = "ApiDestination";
const _ADA = "ApiDestinationArn";
const _ADE = "AccessDeniedException";
const _ADRL = "ApiDestinationResponseList";
const _ADS = "ApiDestinationState";
const _ADp = "ApiDestinations";
const _AE = "AuthorizationEndpoint";
const _AES = "ActivateEventSource";
const _AESR = "ActivateEventSourceRequest";
const _AHPS = "AuthHeaderParametersSensitive";
const _AKAP = "ApiKeyAuthParameters";
const _AKN = "ApiKeyName";
const _AKV = "ApiKeyValue";
const _AN = "ArchiveName";
const _AP = "ArrayProperties";
const _API = "AssignPublicIp";
const _APu = "AuthParameters";
const _ARL = "ArchiveResponseList";
const _ASP = "AppSyncParameters";
const _AT = "AuthorizationType";
const _AVC = "AwsVpcConfiguration";
const _Ac = "Account";
const _Act = "Action";
const _Ar = "Arn";
const _Arc = "Archives";
const _At = "Attempts";
const _BAP = "BatchArrayProperties";
const _BAPa = "BasicAuthParameters";
const _BP = "BatchParameters";
const _BPo = "BodyParameters";
const _BRS = "BatchRetryStrategy";
const _C = "Condition";
const _CA = "ConnectionArn";
const _CAD = "CreateApiDestination";
const _CADR = "CreateApiDestinationRequest";
const _CADRr = "CreateApiDestinationResponse";
const _CAKARP = "ConnectionApiKeyAuthResponseParameters";
const _CAR = "CreateArchiveRequest";
const _CARP = "ConnectionAuthResponseParameters";
const _CARr = "CreateArchiveResponse";
const _CAr = "CreateArchive";
const _CB = "CreatedBy";
const _CBARP = "ConnectionBasicAuthResponseParameters";
const _CBP = "ConnectionBodyParameter";
const _CBPL = "ConnectionBodyParametersList";
const _CC = "CreateConnection";
const _CCAKARP = "CreateConnectionApiKeyAuthRequestParameters";
const _CCARP = "CreateConnectionAuthRequestParameters";
const _CCBARP = "CreateConnectionBasicAuthRequestParameters";
const _CCOACRP = "CreateConnectionOAuthClientRequestParameters";
const _CCOARP = "CreateConnectionOAuthRequestParameters";
const _CCR = "CreateConnectionRequest";
const _CCRr = "CreateConnectionResponse";
const _CE = "CreateEndpoint";
const _CEB = "CreateEventBus";
const _CEBR = "CreateEventBusRequest";
const _CEBRr = "CreateEventBusResponse";
const _CER = "CreateEndpointRequest";
const _CERr = "CreateEndpointResponse";
const _CHP = "ConnectionHeaderParameter";
const _CHPL = "ConnectionHeaderParametersList";
const _CHPo = "ConnectionHttpParameters";
const _CID = "ClientID";
const _CME = "ConcurrentModificationException";
const _COACRP = "ConnectionOAuthClientResponseParameters";
const _COARP = "ConnectionOAuthResponseParameters";
const _CP = "ConnectivityParameters";
const _CPES = "CreatePartnerEventSource";
const _CPESR = "CreatePartnerEventSourceRequest";
const _CPESRr = "CreatePartnerEventSourceResponse";
const _CPS = "CapacityProviderStrategy";
const _CPSI = "CapacityProviderStrategyItem";
const _CPl = "ClientParameters";
const _CQSP = "ConnectionQueryStringParameter";
const _CQSPL = "ConnectionQueryStringParametersList";
const _CR = "CancelReplay";
const _CRCA = "ConnectivityResourceConfigurationArn";
const _CRL = "ConnectionResponseList";
const _CRP = "ConnectivityResourceParameters";
const _CRR = "CancelReplayRequest";
const _CRRa = "CancelReplayResponse";
const _CS = "ConnectionState";
const _CSl = "ClientSecret";
const _CT = "CreationTime";
const _Co = "Connection";
const _Con = "Connections";
const _D = "Description";
const _DA = "DeleteArchive";
const _DAD = "DeleteApiDestination";
const _DADR = "DeleteApiDestinationRequest";
const _DADRe = "DeleteApiDestinationResponse";
const _DADRes = "DescribeApiDestinationRequest";
const _DADResc = "DescribeApiDestinationResponse";
const _DADe = "DescribeApiDestination";
const _DAR = "DeleteArchiveRequest";
const _DARe = "DeleteArchiveResponse";
const _DARes = "DescribeArchiveRequest";
const _DAResc = "DescribeArchiveResponse";
const _DAe = "DescribeArchive";
const _DC = "DeauthorizeConnection";
const _DCCP = "DescribeConnectionConnectivityParameters";
const _DCR = "DeauthorizeConnectionRequest";
const _DCRP = "DescribeConnectionResourceParameters";
const _DCRe = "DeauthorizeConnectionResponse";
const _DCRel = "DeleteConnectionRequest";
const _DCRele = "DeleteConnectionResponse";
const _DCRes = "DescribeConnectionRequest";
const _DCResc = "DescribeConnectionResponse";
const _DCe = "DeleteConnection";
const _DCes = "DescribeConnection";
const _DE = "DeleteEndpoint";
const _DEB = "DeleteEventBus";
const _DEBR = "DeleteEventBusRequest";
const _DEBRe = "DescribeEventBusRequest";
const _DEBRes = "DescribeEventBusResponse";
const _DEBe = "DescribeEventBus";
const _DER = "DeleteEndpointRequest";
const _DERe = "DeleteEndpointResponse";
const _DERes = "DescribeEndpointRequest";
const _DEResc = "DescribeEndpointResponse";
const _DES = "DeactivateEventSource";
const _DESR = "DeactivateEventSourceRequest";
const _DESRe = "DescribeEventSourceRequest";
const _DESRes = "DescribeEventSourceResponse";
const _DESe = "DescribeEventSource";
const _DEe = "DescribeEndpoint";
const _DLC = "DeadLetterConfig";
const _DPES = "DeletePartnerEventSource";
const _DPESR = "DeletePartnerEventSourceRequest";
const _DPESRe = "DescribePartnerEventSourceRequest";
const _DPESRes = "DescribePartnerEventSourceResponse";
const _DPESe = "DescribePartnerEventSource";
const _DR = "DeleteRule";
const _DRR = "DeleteRuleRequest";
const _DRRe = "DescribeReplayRequest";
const _DRRes = "DescribeReplayResponse";
const _DRResc = "DescribeRuleRequest";
const _DRRescr = "DescribeRuleResponse";
const _DRRi = "DisableRuleRequest";
const _DRe = "DescribeReplay";
const _DRes = "DescribeRule";
const _DRi = "DisableRule";
const _DT = "DetailType";
const _DU = "DbUser";
const _Da = "Database";
const _De = "Destination";
const _Det = "Detail";
const _E = "Endpoint";
const _EB = "EventBuses";
const _EBA = "EventBusArn";
const _EBL = "EventBusList";
const _EBN = "EventBusName";
const _EBv = "EventBus";
const _EC = "EventCount";
const _ECr = "ErrorCode";
const _EEB = "EndpointEventBus";
const _EEBL = "EndpointEventBusList";
const _EEC = "EnableExecuteCommand";
const _EECSMT = "EnableECSManagedTags";
const _EET = "EventEndTime";
const _EI = "EndpointId";
const _EIv = "EventId";
const _EL = "EndpointList";
const _ELRT = "EventLastReplayedTime";
const _EM = "ErrorMessage";
const _EP = "EventPattern";
const _EPc = "EcsParameters";
const _ER = "EnableRule";
const _ERR = "EnableRuleRequest";
const _ES = "EventSource";
const _ESA = "EventSourceArn";
const _ESL = "EventSourceList";
const _ESN = "EventSourceName";
const _EST = "EventStartTime";
const _ESv = "EventSources";
const _ET = "ExpirationTime";
const _EU = "EndpointUrl";
const _En = "Endpoints";
const _Ent = "Entries";
const _Ev = "Event";
const _F = "Force";
const _FA = "FilterArns";
const _FC = "FailoverConfig";
const _FE = "FailedEntries";
const _FEC = "FailedEntryCount";
const _G = "Group";
const _GQLO = "GraphQLOperation";
const _HC = "HealthCheck";
const _HM = "HttpMethod";
const _HP = "HeaderParameters";
const _HPt = "HttpParameters";
const _HR = "HomeRegion";
const _HVS = "HeaderValueSensitive";
const _I = "Ids";
const _ICP = "InvocationConnectivityParameters";
const _ID = "IncludeDetail";
const _IE = "InternalException";
const _IEPE = "InvalidEventPatternException";
const _IEn = "InvocationEndpoint";
const _IHP = "InvocationHttpParameters";
const _IP = "InputPath";
const _IPM = "InputPathsMap";
const _IRLPS = "InvocationRateLimitPerSecond";
const _ISE = "IllegalStatusException";
const _ISEn = "InvalidStateException";
const _IT = "InputTransformer";
const _ITn = "InputTemplate";
const _IVS = "IsValueSecret";
const _Id = "Id";
const _In = "Input";
const _JD = "JobDefinition";
const _JN = "JobName";
const _K = "Key";
const _KKI = "KmsKeyIdentifier";
const _KP = "KinesisParameters";
const _L = "Limit";
const _LA = "ListArchives";
const _LAD = "ListApiDestinations";
const _LADR = "ListApiDestinationsRequest";
const _LADRi = "ListApiDestinationsResponse";
const _LAR = "ListArchivesRequest";
const _LARi = "ListArchivesResponse";
const _LAT = "LastAuthorizedTime";
const _LC = "LogConfig";
const _LCR = "ListConnectionsRequest";
const _LCRi = "ListConnectionsResponse";
const _LCi = "ListConnections";
const _LE = "ListEndpoints";
const _LEB = "ListEventBuses";
const _LEBR = "ListEventBusesRequest";
const _LEBRi = "ListEventBusesResponse";
const _LEE = "LimitExceededException";
const _LER = "ListEndpointsRequest";
const _LERi = "ListEndpointsResponse";
const _LES = "ListEventSources";
const _LESR = "ListEventSourcesRequest";
const _LESRi = "ListEventSourcesResponse";
const _LMT = "LastModifiedTime";
const _LPES = "ListPartnerEventSources";
const _LPESA = "ListPartnerEventSourceAccounts";
const _LPESAR = "ListPartnerEventSourceAccountsRequest";
const _LPESARi = "ListPartnerEventSourceAccountsResponse";
const _LPESR = "ListPartnerEventSourcesRequest";
const _LPESRi = "ListPartnerEventSourcesResponse";
const _LR = "ListReplays";
const _LRNBT = "ListRuleNamesByTarget";
const _LRNBTR = "ListRuleNamesByTargetRequest";
const _LRNBTRi = "ListRuleNamesByTargetResponse";
const _LRR = "ListReplaysRequest";
const _LRRi = "ListReplaysResponse";
const _LRRis = "ListRulesRequest";
const _LRRist = "ListRulesResponse";
const _LRi = "ListRules";
const _LT = "LaunchType";
const _LTBR = "ListTargetsByRule";
const _LTBRR = "ListTargetsByRuleRequest";
const _LTBRRi = "ListTargetsByRuleResponse";
const _LTFR = "ListTagsForResource";
const _LTFRR = "ListTagsForResourceRequest";
const _LTFRRi = "ListTagsForResourceResponse";
const _Le = "Level";
const _MB = "ManagedBy";
const _MEAIS = "MaximumEventAgeInSeconds";
const _MGI = "MessageGroupId";
const _MR = "MaxResults";
const _MRA = "MaximumRetryAttempts";
const _MRE = "ManagedRuleException";
const _N = "Name";
const _NC = "NetworkConfiguration";
const _NP = "NamePrefix";
const _NT = "NextToken";
const _OAHP = "OAuthHttpParameters";
const _OAP = "OAuthParameters";
const _ODE = "OperationDisabledException";
const _P = "Password";
const _PC = "PlacementConstraints";
const _PCl = "PlacementConstraint";
const _PE = "PutEvents";
const _PER = "PutEventsRequest";
const _PERE = "PutEventsRequestEntry";
const _PEREL = "PutEventsRequestEntryList";
const _PERELu = "PutEventsResultEntryList";
const _PEREu = "PutEventsResultEntry";
const _PERu = "PutEventsResponse";
const _PES = "PartnerEventSources";
const _PESA = "PartnerEventSourceAccounts";
const _PESAL = "PartnerEventSourceAccountList";
const _PESAa = "PartnerEventSourceAccount";
const _PESL = "PartnerEventSourceList";
const _PESa = "PartnerEventSource";
const _PKP = "PartitionKeyPath";
const _PLEE = "PolicyLengthExceededException";
const _PP = "PutPermission";
const _PPE = "PutPartnerEvents";
const _PPER = "PutPartnerEventsRequest";
const _PPERE = "PutPartnerEventsRequestEntry";
const _PPEREL = "PutPartnerEventsRequestEntryList";
const _PPERELu = "PutPartnerEventsResultEntryList";
const _PPEREu = "PutPartnerEventsResultEntry";
const _PPERu = "PutPartnerEventsResponse";
const _PPL = "PipelineParameterList";
const _PPR = "PutPermissionRequest";
const _PPV = "PathParameterValues";
const _PR = "PutRule";
const _PRR = "PutRuleRequest";
const _PRRu = "PutRuleResponse";
const _PS = "PlacementStrategy";
const _PSl = "PlacementStrategies";
const _PT = "PropagateTags";
const _PTR = "PutTargetsRequest";
const _PTRE = "PutTargetsResultEntry";
const _PTREL = "PutTargetsResultEntryList";
const _PTRu = "PutTargetsResponse";
const _PTu = "PutTargets";
const _PV = "PlatformVersion";
const _Po = "Policy";
const _Pr = "Primary";
const _Pri = "Principal";
const _QSP = "QueryStringParameters";
const _QSVS = "QueryStringValueSensitive";
const _R = "Replays";
const _RA = "ReplayArn";
const _RAA = "ResourceAssociationArn";
const _RAEE = "ResourceAlreadyExistsException";
const _RAP = "RemoveAllPermissions";
const _RARN = "ResourceARN";
const _RAo = "RoleArn";
const _RAu = "RuleArn";
const _RC = "RoutingConfig";
const _RCA = "ResourceConfigurationArn";
const _RCP = "RunCommandParameters";
const _RCT = "RunCommandTargets";
const _RCTu = "RunCommandTarget";
const _RCe = "ReplicationConfig";
const _RD = "RetentionDays";
const _RDP = "RedshiftDataParameters";
const _RDe = "ReplayDestination";
const _RET = "ReplayEndTime";
const _RI = "ReferenceId";
const _RL = "ReplayList";
const _RN = "ReplayName";
const _RNFE = "ResourceNotFoundException";
const _RNu = "RuleNames";
const _RP = "ResourceParameters";
const _RPR = "RemovePermissionRequest";
const _RPe = "RetryPolicy";
const _RPem = "RemovePermission";
const _RRL = "RuleResponseList";
const _RS = "RetryStrategy";
const _RST = "ReplayStartTime";
const _RT = "RemoveTargets";
const _RTR = "RemoveTargetsRequest";
const _RTRE = "RemoveTargetsResultEntry";
const _RTREL = "RemoveTargetsResultEntryList";
const _RTRe = "RemoveTargetsResponse";
const _Re = "Resources";
const _Rep = "Replay";
const _Res = "Result";
const _Ro = "Route";
const _Ru = "Rules";
const _Rul = "Rule";
const _S = "Sql";
const _SA = "SecretArn";
const _SB = "SizeBytes";
const _SE = "ScheduleExpression";
const _SG = "SecurityGroups";
const _SI = "StatementId";
const _SMA = "SecretManagerArn";
const _SMPP = "SageMakerPipelineParameter";
const _SMPPL = "SageMakerPipelineParameterList";
const _SMPPa = "SageMakerPipelineParameters";
const _SN = "StatementName";
const _SP = "SqsParameters";
const _SR = "StateReason";
const _SRR = "StartReplayRequest";
const _SRRt = "StartReplayResponse";
const _SRt = "StartReplay";
const _SS = "SensitiveString";
const _Se = "Secondary";
const _Si = "Size";
const _So = "Source";
const _Sq = "Sqls";
const _St = "State";
const _Su = "Subnets";
const _T = "Type";
const _TA = "TargetArn";
const _TC = "TaskCount";
const _TDA = "TaskDefinitionArn";
const _TE = "ThrottlingException";
const _TEP = "TestEventPattern";
const _TEPR = "TestEventPatternRequest";
const _TEPRe = "TestEventPatternResponse";
const _TH = "TraceHeader";
const _TI = "TargetId";
const _TK = "TagKeys";
const _TL = "TagList";
const _TLa = "TargetList";
const _TR = "TagResource";
const _TRR = "TagResourceRequest";
const _TRRa = "TagResourceResponse";
const _Ta = "Tags";
const _Tag = "Tag";
const _Tar = "Targets";
const _Targ = "Target";
const _Ti = "Time";
const _U = "Username";
const _UA = "UpdateArchive";
const _UAD = "UpdateApiDestination";
const _UADR = "UpdateApiDestinationRequest";
const _UADRp = "UpdateApiDestinationResponse";
const _UAR = "UpdateArchiveRequest";
const _UARp = "UpdateArchiveResponse";
const _UC = "UpdateConnection";
const _UCAKARP = "UpdateConnectionApiKeyAuthRequestParameters";
const _UCARP = "UpdateConnectionAuthRequestParameters";
const _UCBARP = "UpdateConnectionBasicAuthRequestParameters";
const _UCOACRP = "UpdateConnectionOAuthClientRequestParameters";
const _UCOARP = "UpdateConnectionOAuthRequestParameters";
const _UCR = "UpdateConnectionRequest";
const _UCRp = "UpdateConnectionResponse";
const _UE = "UpdateEndpoint";
const _UEB = "UpdateEventBus";
const _UEBR = "UpdateEventBusRequest";
const _UEBRp = "UpdateEventBusResponse";
const _UER = "UpdateEndpointRequest";
const _UERp = "UpdateEndpointResponse";
const _UR = "UntagResource";
const _URR = "UntagResourceRequest";
const _URRn = "UntagResourceResponse";
const _V = "Value";
const _Va = "Values";
const _WE = "WithEvent";
const _aC = "awsvpcConfiguration";
const _b = "base";
const _c = "client";
const _cP = "capacityProvider";
const _e = "error";
const _ex = "expression";
const _f = "field";
const _m = "message";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.eventbridge";
const _se = "server";
const _t = "type";
const _w = "weight";
const n0 = "com.amazonaws.eventbridge";
const _s_registry = TypeRegistry.for(_s);
var EventBridgeServiceException$ = [-3, _s, "EventBridgeServiceException", 0, [], []];
_s_registry.registerError(EventBridgeServiceException$, EventBridgeServiceException);
const n0_registry = TypeRegistry.for(n0);
var AccessDeniedException$ = [-3, n0, _ADE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(AccessDeniedException$, AccessDeniedException);
var ConcurrentModificationException$ = [-3, n0, _CME,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(ConcurrentModificationException$, ConcurrentModificationException);
var IllegalStatusException$ = [-3, n0, _ISE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(IllegalStatusException$, IllegalStatusException);
var InternalException$ = [-3, n0, _IE,
    { [_e]: _se },
    [_m],
    [0]
];
n0_registry.registerError(InternalException$, InternalException);
var InvalidEventPatternException$ = [-3, n0, _IEPE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(InvalidEventPatternException$, InvalidEventPatternException);
var InvalidStateException$ = [-3, n0, _ISEn,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(InvalidStateException$, InvalidStateException);
var LimitExceededException$ = [-3, n0, _LEE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(LimitExceededException$, LimitExceededException);
var ManagedRuleException$ = [-3, n0, _MRE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(ManagedRuleException$, ManagedRuleException);
var OperationDisabledException$ = [-3, n0, _ODE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(OperationDisabledException$, OperationDisabledException);
var PolicyLengthExceededException$ = [-3, n0, _PLEE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(PolicyLengthExceededException$, PolicyLengthExceededException);
var ResourceAlreadyExistsException$ = [-3, n0, _RAEE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(ResourceAlreadyExistsException$, ResourceAlreadyExistsException);
var ResourceNotFoundException$ = [-3, n0, _RNFE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(ResourceNotFoundException$, ResourceNotFoundException);
var ThrottlingException$ = [-3, n0, _TE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(ThrottlingException$, ThrottlingException);
const errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
var AuthHeaderParametersSensitive = [0, n0, _AHPS, 8, 0];
var GraphQLOperation = [0, n0, _GQLO, 8, 0];
var HeaderValueSensitive = [0, n0, _HVS, 8, 0];
var QueryStringValueSensitive = [0, n0, _QSVS, 8, 0];
var SensitiveString = [0, n0, _SS, 8, 0];
var Sql = [0, n0, _S, 8, 0];
var ActivateEventSourceRequest$ = [3, n0, _AESR,
    0,
    [_N],
    [0], 1
];
var ApiDestination$ = [3, n0, _AD,
    0,
    [_ADA, _N, _ADS, _CA, _IEn, _HM, _IRLPS, _CT, _LMT],
    [0, 0, 0, 0, 0, 0, 1, 4, 4]
];
var AppSyncParameters$ = [3, n0, _ASP,
    0,
    [_GQLO],
    [[() => GraphQLOperation, 0]]
];
var Archive$ = [3, n0, _A,
    0,
    [_AN, _ESA, _St, _SR, _RD, _SB, _EC, _CT],
    [0, 0, 0, 0, 1, 1, 1, 4]
];
var AwsVpcConfiguration$ = [3, n0, _AVC,
    0,
    [_Su, _SG, _API],
    [64 | 0, 64 | 0, 0], 1
];
var BatchArrayProperties$ = [3, n0, _BAP,
    0,
    [_Si],
    [1]
];
var BatchParameters$ = [3, n0, _BP,
    0,
    [_JD, _JN, _AP, _RS],
    [0, 0, () => BatchArrayProperties$, () => BatchRetryStrategy$], 2
];
var BatchRetryStrategy$ = [3, n0, _BRS,
    0,
    [_At],
    [1]
];
var CancelReplayRequest$ = [3, n0, _CRR,
    0,
    [_RN],
    [0], 1
];
var CancelReplayResponse$ = [3, n0, _CRRa,
    0,
    [_RA, _St, _SR],
    [0, 0, 0]
];
var CapacityProviderStrategyItem$ = [3, n0, _CPSI,
    0,
    [_cP, _w, _b],
    [0, 1, 1], 1
];
var Condition$ = [3, n0, _C,
    0,
    [_T, _K, _V],
    [0, 0, 0], 3
];
var Connection$ = [3, n0, _Co,
    0,
    [_CA, _N, _CS, _SR, _AT, _CT, _LMT, _LAT],
    [0, 0, 0, 0, 0, 4, 4, 4]
];
var ConnectionApiKeyAuthResponseParameters$ = [3, n0, _CAKARP,
    0,
    [_AKN],
    [0]
];
var ConnectionAuthResponseParameters$ = [3, n0, _CARP,
    0,
    [_BAPa, _OAP, _AKAP, _IHP, _CP],
    [() => ConnectionBasicAuthResponseParameters$, [() => ConnectionOAuthResponseParameters$, 0], () => ConnectionApiKeyAuthResponseParameters$, [() => ConnectionHttpParameters$, 0], () => DescribeConnectionConnectivityParameters$]
];
var ConnectionBasicAuthResponseParameters$ = [3, n0, _CBARP,
    0,
    [_U],
    [0]
];
var ConnectionBodyParameter$ = [3, n0, _CBP,
    0,
    [_K, _V, _IVS],
    [0, [() => SensitiveString, 0], 2]
];
var ConnectionHeaderParameter$ = [3, n0, _CHP,
    0,
    [_K, _V, _IVS],
    [0, [() => HeaderValueSensitive, 0], 2]
];
var ConnectionHttpParameters$ = [3, n0, _CHPo,
    0,
    [_HP, _QSP, _BPo],
    [[() => ConnectionHeaderParametersList, 0], [() => ConnectionQueryStringParametersList, 0], [() => ConnectionBodyParametersList, 0]]
];
var ConnectionOAuthClientResponseParameters$ = [3, n0, _COACRP,
    0,
    [_CID],
    [0]
];
var ConnectionOAuthResponseParameters$ = [3, n0, _COARP,
    0,
    [_CPl, _AE, _HM, _OAHP],
    [() => ConnectionOAuthClientResponseParameters$, 0, 0, [() => ConnectionHttpParameters$, 0]]
];
var ConnectionQueryStringParameter$ = [3, n0, _CQSP,
    0,
    [_K, _V, _IVS],
    [0, [() => QueryStringValueSensitive, 0], 2]
];
var ConnectivityResourceConfigurationArn$ = [3, n0, _CRCA,
    0,
    [_RCA],
    [0], 1
];
var ConnectivityResourceParameters$ = [3, n0, _CRP,
    0,
    [_RP],
    [() => ConnectivityResourceConfigurationArn$], 1
];
var CreateApiDestinationRequest$ = [3, n0, _CADR,
    0,
    [_N, _CA, _IEn, _HM, _D, _IRLPS],
    [0, 0, 0, 0, 0, 1], 4
];
var CreateApiDestinationResponse$ = [3, n0, _CADRr,
    0,
    [_ADA, _ADS, _CT, _LMT],
    [0, 0, 4, 4]
];
var CreateArchiveRequest$ = [3, n0, _CAR,
    0,
    [_AN, _ESA, _D, _EP, _RD, _KKI],
    [0, 0, 0, 0, 1, 0], 2
];
var CreateArchiveResponse$ = [3, n0, _CARr,
    0,
    [_AA, _St, _SR, _CT],
    [0, 0, 0, 4]
];
var CreateConnectionApiKeyAuthRequestParameters$ = [3, n0, _CCAKARP,
    0,
    [_AKN, _AKV],
    [0, [() => AuthHeaderParametersSensitive, 0]], 2
];
var CreateConnectionAuthRequestParameters$ = [3, n0, _CCARP,
    0,
    [_BAPa, _OAP, _AKAP, _IHP, _CP],
    [[() => CreateConnectionBasicAuthRequestParameters$, 0], [() => CreateConnectionOAuthRequestParameters$, 0], [() => CreateConnectionApiKeyAuthRequestParameters$, 0], [() => ConnectionHttpParameters$, 0], () => ConnectivityResourceParameters$]
];
var CreateConnectionBasicAuthRequestParameters$ = [3, n0, _CCBARP,
    0,
    [_U, _P],
    [0, [() => AuthHeaderParametersSensitive, 0]], 2
];
var CreateConnectionOAuthClientRequestParameters$ = [3, n0, _CCOACRP,
    0,
    [_CID, _CSl],
    [0, [() => AuthHeaderParametersSensitive, 0]], 2
];
var CreateConnectionOAuthRequestParameters$ = [3, n0, _CCOARP,
    0,
    [_CPl, _AE, _HM, _OAHP],
    [[() => CreateConnectionOAuthClientRequestParameters$, 0], 0, 0, [() => ConnectionHttpParameters$, 0]], 3
];
var CreateConnectionRequest$ = [3, n0, _CCR,
    0,
    [_N, _AT, _APu, _D, _ICP, _KKI],
    [0, 0, [() => CreateConnectionAuthRequestParameters$, 0], 0, () => ConnectivityResourceParameters$, 0], 3
];
var CreateConnectionResponse$ = [3, n0, _CCRr,
    0,
    [_CA, _CS, _CT, _LMT],
    [0, 0, 4, 4]
];
var CreateEndpointRequest$ = [3, n0, _CER,
    0,
    [_N, _RC, _EB, _D, _RCe, _RAo],
    [0, () => RoutingConfig$, () => EndpointEventBusList, 0, () => ReplicationConfig$, 0], 3
];
var CreateEndpointResponse$ = [3, n0, _CERr,
    0,
    [_N, _Ar, _RC, _RCe, _EB, _RAo, _St],
    [0, 0, () => RoutingConfig$, () => ReplicationConfig$, () => EndpointEventBusList, 0, 0]
];
var CreateEventBusRequest$ = [3, n0, _CEBR,
    0,
    [_N, _ESN, _D, _KKI, _DLC, _LC, _Ta],
    [0, 0, 0, 0, () => DeadLetterConfig$, () => LogConfig$, () => TagList], 1
];
var CreateEventBusResponse$ = [3, n0, _CEBRr,
    0,
    [_EBA, _D, _KKI, _DLC, _LC],
    [0, 0, 0, () => DeadLetterConfig$, () => LogConfig$]
];
var CreatePartnerEventSourceRequest$ = [3, n0, _CPESR,
    0,
    [_N, _Ac],
    [0, 0], 2
];
var CreatePartnerEventSourceResponse$ = [3, n0, _CPESRr,
    0,
    [_ESA],
    [0]
];
var DeactivateEventSourceRequest$ = [3, n0, _DESR,
    0,
    [_N],
    [0], 1
];
var DeadLetterConfig$ = [3, n0, _DLC,
    0,
    [_Ar],
    [0]
];
var DeauthorizeConnectionRequest$ = [3, n0, _DCR,
    0,
    [_N],
    [0], 1
];
var DeauthorizeConnectionResponse$ = [3, n0, _DCRe,
    0,
    [_CA, _CS, _CT, _LMT, _LAT],
    [0, 0, 4, 4, 4]
];
var DeleteApiDestinationRequest$ = [3, n0, _DADR,
    0,
    [_N],
    [0], 1
];
var DeleteApiDestinationResponse$ = [3, n0, _DADRe,
    0,
    [],
    []
];
var DeleteArchiveRequest$ = [3, n0, _DAR,
    0,
    [_AN],
    [0], 1
];
var DeleteArchiveResponse$ = [3, n0, _DARe,
    0,
    [],
    []
];
var DeleteConnectionRequest$ = [3, n0, _DCRel,
    0,
    [_N],
    [0], 1
];
var DeleteConnectionResponse$ = [3, n0, _DCRele,
    0,
    [_CA, _CS, _CT, _LMT, _LAT],
    [0, 0, 4, 4, 4]
];
var DeleteEndpointRequest$ = [3, n0, _DER,
    0,
    [_N],
    [0], 1
];
var DeleteEndpointResponse$ = [3, n0, _DERe,
    0,
    [],
    []
];
var DeleteEventBusRequest$ = [3, n0, _DEBR,
    0,
    [_N],
    [0], 1
];
var DeletePartnerEventSourceRequest$ = [3, n0, _DPESR,
    0,
    [_N, _Ac],
    [0, 0], 2
];
var DeleteRuleRequest$ = [3, n0, _DRR,
    0,
    [_N, _EBN, _F],
    [0, 0, 2], 1
];
var DescribeApiDestinationRequest$ = [3, n0, _DADRes,
    0,
    [_N],
    [0], 1
];
var DescribeApiDestinationResponse$ = [3, n0, _DADResc,
    0,
    [_ADA, _N, _D, _ADS, _CA, _IEn, _HM, _IRLPS, _CT, _LMT],
    [0, 0, 0, 0, 0, 0, 0, 1, 4, 4]
];
var DescribeArchiveRequest$ = [3, n0, _DARes,
    0,
    [_AN],
    [0], 1
];
var DescribeArchiveResponse$ = [3, n0, _DAResc,
    0,
    [_AA, _AN, _ESA, _D, _EP, _St, _SR, _KKI, _RD, _SB, _EC, _CT],
    [0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 4]
];
var DescribeConnectionConnectivityParameters$ = [3, n0, _DCCP,
    0,
    [_RP],
    [() => DescribeConnectionResourceParameters$], 1
];
var DescribeConnectionRequest$ = [3, n0, _DCRes,
    0,
    [_N],
    [0], 1
];
var DescribeConnectionResourceParameters$ = [3, n0, _DCRP,
    0,
    [_RCA, _RAA],
    [0, 0], 2
];
var DescribeConnectionResponse$ = [3, n0, _DCResc,
    0,
    [_CA, _N, _D, _ICP, _CS, _SR, _AT, _SA, _KKI, _APu, _CT, _LMT, _LAT],
    [0, 0, 0, () => DescribeConnectionConnectivityParameters$, 0, 0, 0, 0, 0, [() => ConnectionAuthResponseParameters$, 0], 4, 4, 4]
];
var DescribeEndpointRequest$ = [3, n0, _DERes,
    0,
    [_N, _HR],
    [0, 0], 1
];
var DescribeEndpointResponse$ = [3, n0, _DEResc,
    0,
    [_N, _D, _Ar, _RC, _RCe, _EB, _RAo, _EI, _EU, _St, _SR, _CT, _LMT],
    [0, 0, 0, () => RoutingConfig$, () => ReplicationConfig$, () => EndpointEventBusList, 0, 0, 0, 0, 0, 4, 4]
];
var DescribeEventBusRequest$ = [3, n0, _DEBRe,
    0,
    [_N],
    [0]
];
var DescribeEventBusResponse$ = [3, n0, _DEBRes,
    0,
    [_N, _Ar, _D, _KKI, _DLC, _Po, _LC, _CT, _LMT],
    [0, 0, 0, 0, () => DeadLetterConfig$, 0, () => LogConfig$, 4, 4]
];
var DescribeEventSourceRequest$ = [3, n0, _DESRe,
    0,
    [_N],
    [0], 1
];
var DescribeEventSourceResponse$ = [3, n0, _DESRes,
    0,
    [_Ar, _CB, _CT, _ET, _N, _St],
    [0, 0, 4, 4, 0, 0]
];
var DescribePartnerEventSourceRequest$ = [3, n0, _DPESRe,
    0,
    [_N],
    [0], 1
];
var DescribePartnerEventSourceResponse$ = [3, n0, _DPESRes,
    0,
    [_Ar, _N],
    [0, 0]
];
var DescribeReplayRequest$ = [3, n0, _DRRe,
    0,
    [_RN],
    [0], 1
];
var DescribeReplayResponse$ = [3, n0, _DRRes,
    0,
    [_RN, _RA, _D, _St, _SR, _ESA, _De, _EST, _EET, _ELRT, _RST, _RET],
    [0, 0, 0, 0, 0, 0, () => ReplayDestination$, 4, 4, 4, 4, 4]
];
var DescribeRuleRequest$ = [3, n0, _DRResc,
    0,
    [_N, _EBN],
    [0, 0], 1
];
var DescribeRuleResponse$ = [3, n0, _DRRescr,
    0,
    [_N, _Ar, _EP, _SE, _St, _D, _RAo, _MB, _EBN, _CB],
    [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
];
var DisableRuleRequest$ = [3, n0, _DRRi,
    0,
    [_N, _EBN],
    [0, 0], 1
];
var EcsParameters$ = [3, n0, _EPc,
    0,
    [_TDA, _TC, _LT, _NC, _PV, _G, _CPS, _EECSMT, _EEC, _PC, _PS, _PT, _RI, _Ta],
    [0, 1, 0, () => NetworkConfiguration$, 0, 0, () => CapacityProviderStrategy, 2, 2, () => PlacementConstraints, () => PlacementStrategies, 0, 0, () => TagList], 1
];
var EnableRuleRequest$ = [3, n0, _ERR,
    0,
    [_N, _EBN],
    [0, 0], 1
];
var Endpoint$ = [3, n0, _E,
    0,
    [_N, _D, _Ar, _RC, _RCe, _EB, _RAo, _EI, _EU, _St, _SR, _CT, _LMT],
    [0, 0, 0, () => RoutingConfig$, () => ReplicationConfig$, () => EndpointEventBusList, 0, 0, 0, 0, 0, 4, 4]
];
var EndpointEventBus$ = [3, n0, _EEB,
    0,
    [_EBA],
    [0], 1
];
var EventBus$ = [3, n0, _EBv,
    0,
    [_N, _Ar, _D, _Po, _CT, _LMT],
    [0, 0, 0, 0, 4, 4]
];
var EventSource$ = [3, n0, _ES,
    0,
    [_Ar, _CB, _CT, _ET, _N, _St],
    [0, 0, 4, 4, 0, 0]
];
var FailoverConfig$ = [3, n0, _FC,
    0,
    [_Pr, _Se],
    [() => Primary$, () => Secondary$], 2
];
var HttpParameters$ = [3, n0, _HPt,
    0,
    [_PPV, _HP, _QSP],
    [64 | 0, 128 | 0, 128 | 0]
];
var InputTransformer$ = [3, n0, _IT,
    0,
    [_ITn, _IPM],
    [0, 128 | 0], 1
];
var KinesisParameters$ = [3, n0, _KP,
    0,
    [_PKP],
    [0], 1
];
var ListApiDestinationsRequest$ = [3, n0, _LADR,
    0,
    [_NP, _CA, _NT, _L],
    [0, 0, 0, 1]
];
var ListApiDestinationsResponse$ = [3, n0, _LADRi,
    0,
    [_ADp, _NT],
    [() => ApiDestinationResponseList, 0]
];
var ListArchivesRequest$ = [3, n0, _LAR,
    0,
    [_NP, _ESA, _St, _NT, _L],
    [0, 0, 0, 0, 1]
];
var ListArchivesResponse$ = [3, n0, _LARi,
    0,
    [_Arc, _NT],
    [() => ArchiveResponseList, 0]
];
var ListConnectionsRequest$ = [3, n0, _LCR,
    0,
    [_NP, _CS, _NT, _L],
    [0, 0, 0, 1]
];
var ListConnectionsResponse$ = [3, n0, _LCRi,
    0,
    [_Con, _NT],
    [() => ConnectionResponseList, 0]
];
var ListEndpointsRequest$ = [3, n0, _LER,
    0,
    [_NP, _HR, _NT, _MR],
    [0, 0, 0, 1]
];
var ListEndpointsResponse$ = [3, n0, _LERi,
    0,
    [_En, _NT],
    [() => EndpointList, 0]
];
var ListEventBusesRequest$ = [3, n0, _LEBR,
    0,
    [_NP, _NT, _L],
    [0, 0, 1]
];
var ListEventBusesResponse$ = [3, n0, _LEBRi,
    0,
    [_EB, _NT],
    [() => EventBusList, 0]
];
var ListEventSourcesRequest$ = [3, n0, _LESR,
    0,
    [_NP, _NT, _L],
    [0, 0, 1]
];
var ListEventSourcesResponse$ = [3, n0, _LESRi,
    0,
    [_ESv, _NT],
    [() => EventSourceList, 0]
];
var ListPartnerEventSourceAccountsRequest$ = [3, n0, _LPESAR,
    0,
    [_ESN, _NT, _L],
    [0, 0, 1], 1
];
var ListPartnerEventSourceAccountsResponse$ = [3, n0, _LPESARi,
    0,
    [_PESA, _NT],
    [() => PartnerEventSourceAccountList, 0]
];
var ListPartnerEventSourcesRequest$ = [3, n0, _LPESR,
    0,
    [_NP, _NT, _L],
    [0, 0, 1], 1
];
var ListPartnerEventSourcesResponse$ = [3, n0, _LPESRi,
    0,
    [_PES, _NT],
    [() => PartnerEventSourceList, 0]
];
var ListReplaysRequest$ = [3, n0, _LRR,
    0,
    [_NP, _St, _ESA, _NT, _L],
    [0, 0, 0, 0, 1]
];
var ListReplaysResponse$ = [3, n0, _LRRi,
    0,
    [_R, _NT],
    [() => ReplayList, 0]
];
var ListRuleNamesByTargetRequest$ = [3, n0, _LRNBTR,
    0,
    [_TA, _EBN, _NT, _L],
    [0, 0, 0, 1], 1
];
var ListRuleNamesByTargetResponse$ = [3, n0, _LRNBTRi,
    0,
    [_RNu, _NT],
    [64 | 0, 0]
];
var ListRulesRequest$ = [3, n0, _LRRis,
    0,
    [_NP, _EBN, _NT, _L],
    [0, 0, 0, 1]
];
var ListRulesResponse$ = [3, n0, _LRRist,
    0,
    [_Ru, _NT],
    [() => RuleResponseList, 0]
];
var ListTagsForResourceRequest$ = [3, n0, _LTFRR,
    0,
    [_RARN],
    [0], 1
];
var ListTagsForResourceResponse$ = [3, n0, _LTFRRi,
    0,
    [_Ta],
    [() => TagList]
];
var ListTargetsByRuleRequest$ = [3, n0, _LTBRR,
    0,
    [_Rul, _EBN, _NT, _L],
    [0, 0, 0, 1], 1
];
var ListTargetsByRuleResponse$ = [3, n0, _LTBRRi,
    0,
    [_Tar, _NT],
    [[() => TargetList, 0], 0]
];
var LogConfig$ = [3, n0, _LC,
    0,
    [_ID, _Le],
    [0, 0]
];
var NetworkConfiguration$ = [3, n0, _NC,
    0,
    [_aC],
    [() => AwsVpcConfiguration$]
];
var PartnerEventSource$ = [3, n0, _PESa,
    0,
    [_Ar, _N],
    [0, 0]
];
var PartnerEventSourceAccount$ = [3, n0, _PESAa,
    0,
    [_Ac, _CT, _ET, _St],
    [0, 4, 4, 0]
];
var PlacementConstraint$ = [3, n0, _PCl,
    0,
    [_t, _ex],
    [0, 0]
];
var PlacementStrategy$ = [3, n0, _PS,
    0,
    [_t, _f],
    [0, 0]
];
var Primary$ = [3, n0, _Pr,
    0,
    [_HC],
    [0], 1
];
var PutEventsRequest$ = [3, n0, _PER,
    0,
    [_Ent, _EI],
    [() => PutEventsRequestEntryList, 0], 1
];
var PutEventsRequestEntry$ = [3, n0, _PERE,
    0,
    [_Ti, _So, _Re, _DT, _Det, _EBN, _TH],
    [4, 0, 64 | 0, 0, 0, 0, 0]
];
var PutEventsResponse$ = [3, n0, _PERu,
    0,
    [_FEC, _Ent],
    [1, () => PutEventsResultEntryList]
];
var PutEventsResultEntry$ = [3, n0, _PEREu,
    0,
    [_EIv, _ECr, _EM],
    [0, 0, 0]
];
var PutPartnerEventsRequest$ = [3, n0, _PPER,
    0,
    [_Ent],
    [() => PutPartnerEventsRequestEntryList], 1
];
var PutPartnerEventsRequestEntry$ = [3, n0, _PPERE,
    0,
    [_Ti, _So, _Re, _DT, _Det],
    [4, 0, 64 | 0, 0, 0]
];
var PutPartnerEventsResponse$ = [3, n0, _PPERu,
    0,
    [_FEC, _Ent],
    [1, () => PutPartnerEventsResultEntryList]
];
var PutPartnerEventsResultEntry$ = [3, n0, _PPEREu,
    0,
    [_EIv, _ECr, _EM],
    [0, 0, 0]
];
var PutPermissionRequest$ = [3, n0, _PPR,
    0,
    [_EBN, _Act, _Pri, _SI, _C, _Po],
    [0, 0, 0, 0, () => Condition$, 0]
];
var PutRuleRequest$ = [3, n0, _PRR,
    0,
    [_N, _SE, _EP, _St, _D, _RAo, _Ta, _EBN],
    [0, 0, 0, 0, 0, 0, () => TagList, 0], 1
];
var PutRuleResponse$ = [3, n0, _PRRu,
    0,
    [_RAu],
    [0]
];
var PutTargetsRequest$ = [3, n0, _PTR,
    0,
    [_Rul, _Tar, _EBN],
    [0, [() => TargetList, 0], 0], 2
];
var PutTargetsResponse$ = [3, n0, _PTRu,
    0,
    [_FEC, _FE],
    [1, () => PutTargetsResultEntryList]
];
var PutTargetsResultEntry$ = [3, n0, _PTRE,
    0,
    [_TI, _ECr, _EM],
    [0, 0, 0]
];
var RedshiftDataParameters$ = [3, n0, _RDP,
    0,
    [_Da, _SMA, _DU, _S, _SN, _WE, _Sq],
    [0, 0, 0, [() => Sql, 0], 0, 2, [() => Sqls, 0]], 1
];
var RemovePermissionRequest$ = [3, n0, _RPR,
    0,
    [_SI, _RAP, _EBN],
    [0, 2, 0]
];
var RemoveTargetsRequest$ = [3, n0, _RTR,
    0,
    [_Rul, _I, _EBN, _F],
    [0, 64 | 0, 0, 2], 2
];
var RemoveTargetsResponse$ = [3, n0, _RTRe,
    0,
    [_FEC, _FE],
    [1, () => RemoveTargetsResultEntryList]
];
var RemoveTargetsResultEntry$ = [3, n0, _RTRE,
    0,
    [_TI, _ECr, _EM],
    [0, 0, 0]
];
var Replay$ = [3, n0, _Rep,
    0,
    [_RN, _ESA, _St, _SR, _EST, _EET, _ELRT, _RST, _RET],
    [0, 0, 0, 0, 4, 4, 4, 4, 4]
];
var ReplayDestination$ = [3, n0, _RDe,
    0,
    [_Ar, _FA],
    [0, 64 | 0], 1
];
var ReplicationConfig$ = [3, n0, _RCe,
    0,
    [_St],
    [0]
];
var RetryPolicy$ = [3, n0, _RPe,
    0,
    [_MRA, _MEAIS],
    [1, 1]
];
var RoutingConfig$ = [3, n0, _RC,
    0,
    [_FC],
    [() => FailoverConfig$], 1
];
var Rule$ = [3, n0, _Rul,
    0,
    [_N, _Ar, _EP, _St, _D, _SE, _RAo, _MB, _EBN],
    [0, 0, 0, 0, 0, 0, 0, 0, 0]
];
var RunCommandParameters$ = [3, n0, _RCP,
    0,
    [_RCT],
    [() => RunCommandTargets], 1
];
var RunCommandTarget$ = [3, n0, _RCTu,
    0,
    [_K, _Va],
    [0, 64 | 0], 2
];
var SageMakerPipelineParameter$ = [3, n0, _SMPP,
    0,
    [_N, _V],
    [0, 0], 2
];
var SageMakerPipelineParameters$ = [3, n0, _SMPPa,
    0,
    [_PPL],
    [() => SageMakerPipelineParameterList]
];
var Secondary$ = [3, n0, _Se,
    0,
    [_Ro],
    [0], 1
];
var SqsParameters$ = [3, n0, _SP,
    0,
    [_MGI],
    [0]
];
var StartReplayRequest$ = [3, n0, _SRR,
    0,
    [_RN, _ESA, _EST, _EET, _De, _D],
    [0, 0, 4, 4, () => ReplayDestination$, 0], 5
];
var StartReplayResponse$ = [3, n0, _SRRt,
    0,
    [_RA, _St, _SR, _RST],
    [0, 0, 0, 4]
];
var Tag$ = [3, n0, _Tag,
    0,
    [_K, _V],
    [0, 0], 2
];
var TagResourceRequest$ = [3, n0, _TRR,
    0,
    [_RARN, _Ta],
    [0, () => TagList], 2
];
var TagResourceResponse$ = [3, n0, _TRRa,
    0,
    [],
    []
];
var Target$ = [3, n0, _Targ,
    0,
    [_Id, _Ar, _RAo, _In, _IP, _IT, _KP, _RCP, _EPc, _BP, _SP, _HPt, _RDP, _SMPPa, _DLC, _RPe, _ASP],
    [0, 0, 0, 0, 0, () => InputTransformer$, () => KinesisParameters$, () => RunCommandParameters$, () => EcsParameters$, () => BatchParameters$, () => SqsParameters$, () => HttpParameters$, [() => RedshiftDataParameters$, 0], () => SageMakerPipelineParameters$, () => DeadLetterConfig$, () => RetryPolicy$, [() => AppSyncParameters$, 0]], 2
];
var TestEventPatternRequest$ = [3, n0, _TEPR,
    0,
    [_EP, _Ev],
    [0, 0], 2
];
var TestEventPatternResponse$ = [3, n0, _TEPRe,
    0,
    [_Res],
    [2]
];
var UntagResourceRequest$ = [3, n0, _URR,
    0,
    [_RARN, _TK],
    [0, 64 | 0], 2
];
var UntagResourceResponse$ = [3, n0, _URRn,
    0,
    [],
    []
];
var UpdateApiDestinationRequest$ = [3, n0, _UADR,
    0,
    [_N, _D, _CA, _IEn, _HM, _IRLPS],
    [0, 0, 0, 0, 0, 1], 1
];
var UpdateApiDestinationResponse$ = [3, n0, _UADRp,
    0,
    [_ADA, _ADS, _CT, _LMT],
    [0, 0, 4, 4]
];
var UpdateArchiveRequest$ = [3, n0, _UAR,
    0,
    [_AN, _D, _EP, _RD, _KKI],
    [0, 0, 0, 1, 0], 1
];
var UpdateArchiveResponse$ = [3, n0, _UARp,
    0,
    [_AA, _St, _SR, _CT],
    [0, 0, 0, 4]
];
var UpdateConnectionApiKeyAuthRequestParameters$ = [3, n0, _UCAKARP,
    0,
    [_AKN, _AKV],
    [0, [() => AuthHeaderParametersSensitive, 0]]
];
var UpdateConnectionAuthRequestParameters$ = [3, n0, _UCARP,
    0,
    [_BAPa, _OAP, _AKAP, _IHP, _CP],
    [[() => UpdateConnectionBasicAuthRequestParameters$, 0], [() => UpdateConnectionOAuthRequestParameters$, 0], [() => UpdateConnectionApiKeyAuthRequestParameters$, 0], [() => ConnectionHttpParameters$, 0], () => ConnectivityResourceParameters$]
];
var UpdateConnectionBasicAuthRequestParameters$ = [3, n0, _UCBARP,
    0,
    [_U, _P],
    [0, [() => AuthHeaderParametersSensitive, 0]]
];
var UpdateConnectionOAuthClientRequestParameters$ = [3, n0, _UCOACRP,
    0,
    [_CID, _CSl],
    [0, [() => AuthHeaderParametersSensitive, 0]]
];
var UpdateConnectionOAuthRequestParameters$ = [3, n0, _UCOARP,
    0,
    [_CPl, _AE, _HM, _OAHP],
    [[() => UpdateConnectionOAuthClientRequestParameters$, 0], 0, 0, [() => ConnectionHttpParameters$, 0]]
];
var UpdateConnectionRequest$ = [3, n0, _UCR,
    0,
    [_N, _D, _AT, _APu, _ICP, _KKI],
    [0, 0, 0, [() => UpdateConnectionAuthRequestParameters$, 0], () => ConnectivityResourceParameters$, 0], 1
];
var UpdateConnectionResponse$ = [3, n0, _UCRp,
    0,
    [_CA, _CS, _CT, _LMT, _LAT],
    [0, 0, 4, 4, 4]
];
var UpdateEndpointRequest$ = [3, n0, _UER,
    0,
    [_N, _D, _RC, _RCe, _EB, _RAo],
    [0, 0, () => RoutingConfig$, () => ReplicationConfig$, () => EndpointEventBusList, 0], 1
];
var UpdateEndpointResponse$ = [3, n0, _UERp,
    0,
    [_N, _Ar, _RC, _RCe, _EB, _RAo, _EI, _EU, _St],
    [0, 0, () => RoutingConfig$, () => ReplicationConfig$, () => EndpointEventBusList, 0, 0, 0, 0]
];
var UpdateEventBusRequest$ = [3, n0, _UEBR,
    0,
    [_N, _KKI, _D, _DLC, _LC],
    [0, 0, 0, () => DeadLetterConfig$, () => LogConfig$]
];
var UpdateEventBusResponse$ = [3, n0, _UEBRp,
    0,
    [_Ar, _N, _KKI, _D, _DLC, _LC],
    [0, 0, 0, 0, () => DeadLetterConfig$, () => LogConfig$]
];
var __Unit = "unit";
var ApiDestinationResponseList = [1, n0, _ADRL,
    0, () => ApiDestination$
];
var ArchiveResponseList = [1, n0, _ARL,
    0, () => Archive$
];
var CapacityProviderStrategy = [1, n0, _CPS,
    0, () => CapacityProviderStrategyItem$
];
var ConnectionBodyParametersList = [1, n0, _CBPL,
    0, [() => ConnectionBodyParameter$,
        0]
];
var ConnectionHeaderParametersList = [1, n0, _CHPL,
    0, [() => ConnectionHeaderParameter$,
        0]
];
var ConnectionQueryStringParametersList = [1, n0, _CQSPL,
    0, [() => ConnectionQueryStringParameter$,
        0]
];
var ConnectionResponseList = [1, n0, _CRL,
    0, () => Connection$
];
var EndpointEventBusList = [1, n0, _EEBL,
    0, () => EndpointEventBus$
];
var EndpointList = [1, n0, _EL,
    0, () => Endpoint$
];
var EventBusList = [1, n0, _EBL,
    0, () => EventBus$
];
var EventSourceList = [1, n0, _ESL,
    0, () => EventSource$
];
var PartnerEventSourceAccountList = [1, n0, _PESAL,
    0, () => PartnerEventSourceAccount$
];
var PartnerEventSourceList = [1, n0, _PESL,
    0, () => PartnerEventSource$
];
var PlacementConstraints = [1, n0, _PC,
    0, () => PlacementConstraint$
];
var PlacementStrategies = [1, n0, _PSl,
    0, () => PlacementStrategy$
];
var PutEventsRequestEntryList = [1, n0, _PEREL,
    0, () => PutEventsRequestEntry$
];
var PutEventsResultEntryList = [1, n0, _PERELu,
    0, () => PutEventsResultEntry$
];
var PutPartnerEventsRequestEntryList = [1, n0, _PPEREL,
    0, () => PutPartnerEventsRequestEntry$
];
var PutPartnerEventsResultEntryList = [1, n0, _PPERELu,
    0, () => PutPartnerEventsResultEntry$
];
var PutTargetsResultEntryList = [1, n0, _PTREL,
    0, () => PutTargetsResultEntry$
];
var RemoveTargetsResultEntryList = [1, n0, _RTREL,
    0, () => RemoveTargetsResultEntry$
];
var ReplayList = [1, n0, _RL,
    0, () => Replay$
];
var RuleResponseList = [1, n0, _RRL,
    0, () => Rule$
];
var RunCommandTargets = [1, n0, _RCT,
    0, () => RunCommandTarget$
];
var SageMakerPipelineParameterList = [1, n0, _SMPPL,
    0, () => SageMakerPipelineParameter$
];
var Sqls = [1, n0, _Sq,
    8, [() => Sql,
        0]
];
var TagList = [1, n0, _TL,
    0, () => Tag$
];
var TargetList = [1, n0, _TLa,
    0, [() => Target$,
        0]
];
var ActivateEventSource$ = [9, n0, _AES,
    0, () => ActivateEventSourceRequest$, () => __Unit
];
var CancelReplay$ = [9, n0, _CR,
    0, () => CancelReplayRequest$, () => CancelReplayResponse$
];
var CreateApiDestination$ = [9, n0, _CAD,
    0, () => CreateApiDestinationRequest$, () => CreateApiDestinationResponse$
];
var CreateArchive$ = [9, n0, _CAr,
    0, () => CreateArchiveRequest$, () => CreateArchiveResponse$
];
var CreateConnection$ = [9, n0, _CC,
    0, () => CreateConnectionRequest$, () => CreateConnectionResponse$
];
var CreateEndpoint$ = [9, n0, _CE,
    0, () => CreateEndpointRequest$, () => CreateEndpointResponse$
];
var CreateEventBus$ = [9, n0, _CEB,
    0, () => CreateEventBusRequest$, () => CreateEventBusResponse$
];
var CreatePartnerEventSource$ = [9, n0, _CPES,
    0, () => CreatePartnerEventSourceRequest$, () => CreatePartnerEventSourceResponse$
];
var DeactivateEventSource$ = [9, n0, _DES,
    0, () => DeactivateEventSourceRequest$, () => __Unit
];
var DeauthorizeConnection$ = [9, n0, _DC,
    0, () => DeauthorizeConnectionRequest$, () => DeauthorizeConnectionResponse$
];
var DeleteApiDestination$ = [9, n0, _DAD,
    0, () => DeleteApiDestinationRequest$, () => DeleteApiDestinationResponse$
];
var DeleteArchive$ = [9, n0, _DA,
    0, () => DeleteArchiveRequest$, () => DeleteArchiveResponse$
];
var DeleteConnection$ = [9, n0, _DCe,
    0, () => DeleteConnectionRequest$, () => DeleteConnectionResponse$
];
var DeleteEndpoint$ = [9, n0, _DE,
    0, () => DeleteEndpointRequest$, () => DeleteEndpointResponse$
];
var DeleteEventBus$ = [9, n0, _DEB,
    0, () => DeleteEventBusRequest$, () => __Unit
];
var DeletePartnerEventSource$ = [9, n0, _DPES,
    0, () => DeletePartnerEventSourceRequest$, () => __Unit
];
var DeleteRule$ = [9, n0, _DR,
    0, () => DeleteRuleRequest$, () => __Unit
];
var DescribeApiDestination$ = [9, n0, _DADe,
    0, () => DescribeApiDestinationRequest$, () => DescribeApiDestinationResponse$
];
var DescribeArchive$ = [9, n0, _DAe,
    0, () => DescribeArchiveRequest$, () => DescribeArchiveResponse$
];
var DescribeConnection$ = [9, n0, _DCes,
    0, () => DescribeConnectionRequest$, () => DescribeConnectionResponse$
];
var DescribeEndpoint$ = [9, n0, _DEe,
    0, () => DescribeEndpointRequest$, () => DescribeEndpointResponse$
];
var DescribeEventBus$ = [9, n0, _DEBe,
    0, () => DescribeEventBusRequest$, () => DescribeEventBusResponse$
];
var DescribeEventSource$ = [9, n0, _DESe,
    0, () => DescribeEventSourceRequest$, () => DescribeEventSourceResponse$
];
var DescribePartnerEventSource$ = [9, n0, _DPESe,
    0, () => DescribePartnerEventSourceRequest$, () => DescribePartnerEventSourceResponse$
];
var DescribeReplay$ = [9, n0, _DRe,
    0, () => DescribeReplayRequest$, () => DescribeReplayResponse$
];
var DescribeRule$ = [9, n0, _DRes,
    0, () => DescribeRuleRequest$, () => DescribeRuleResponse$
];
var DisableRule$ = [9, n0, _DRi,
    0, () => DisableRuleRequest$, () => __Unit
];
var EnableRule$ = [9, n0, _ER,
    0, () => EnableRuleRequest$, () => __Unit
];
var ListApiDestinations$ = [9, n0, _LAD,
    0, () => ListApiDestinationsRequest$, () => ListApiDestinationsResponse$
];
var ListArchives$ = [9, n0, _LA,
    0, () => ListArchivesRequest$, () => ListArchivesResponse$
];
var ListConnections$ = [9, n0, _LCi,
    0, () => ListConnectionsRequest$, () => ListConnectionsResponse$
];
var ListEndpoints$ = [9, n0, _LE,
    0, () => ListEndpointsRequest$, () => ListEndpointsResponse$
];
var ListEventBuses$ = [9, n0, _LEB,
    0, () => ListEventBusesRequest$, () => ListEventBusesResponse$
];
var ListEventSources$ = [9, n0, _LES,
    0, () => ListEventSourcesRequest$, () => ListEventSourcesResponse$
];
var ListPartnerEventSourceAccounts$ = [9, n0, _LPESA,
    0, () => ListPartnerEventSourceAccountsRequest$, () => ListPartnerEventSourceAccountsResponse$
];
var ListPartnerEventSources$ = [9, n0, _LPES,
    0, () => ListPartnerEventSourcesRequest$, () => ListPartnerEventSourcesResponse$
];
var ListReplays$ = [9, n0, _LR,
    0, () => ListReplaysRequest$, () => ListReplaysResponse$
];
var ListRuleNamesByTarget$ = [9, n0, _LRNBT,
    0, () => ListRuleNamesByTargetRequest$, () => ListRuleNamesByTargetResponse$
];
var ListRules$ = [9, n0, _LRi,
    0, () => ListRulesRequest$, () => ListRulesResponse$
];
var ListTagsForResource$ = [9, n0, _LTFR,
    0, () => ListTagsForResourceRequest$, () => ListTagsForResourceResponse$
];
var ListTargetsByRule$ = [9, n0, _LTBR,
    0, () => ListTargetsByRuleRequest$, () => ListTargetsByRuleResponse$
];
var PutEvents$ = [9, n0, _PE,
    0, () => PutEventsRequest$, () => PutEventsResponse$
];
var PutPartnerEvents$ = [9, n0, _PPE,
    0, () => PutPartnerEventsRequest$, () => PutPartnerEventsResponse$
];
var PutPermission$ = [9, n0, _PP,
    0, () => PutPermissionRequest$, () => __Unit
];
var PutRule$ = [9, n0, _PR,
    0, () => PutRuleRequest$, () => PutRuleResponse$
];
var PutTargets$ = [9, n0, _PTu,
    0, () => PutTargetsRequest$, () => PutTargetsResponse$
];
var RemovePermission$ = [9, n0, _RPem,
    0, () => RemovePermissionRequest$, () => __Unit
];
var RemoveTargets$ = [9, n0, _RT,
    0, () => RemoveTargetsRequest$, () => RemoveTargetsResponse$
];
var StartReplay$ = [9, n0, _SRt,
    0, () => StartReplayRequest$, () => StartReplayResponse$
];
var TagResource$ = [9, n0, _TR,
    0, () => TagResourceRequest$, () => TagResourceResponse$
];
var TestEventPattern$ = [9, n0, _TEP,
    0, () => TestEventPatternRequest$, () => TestEventPatternResponse$
];
var UntagResource$ = [9, n0, _UR,
    0, () => UntagResourceRequest$, () => UntagResourceResponse$
];
var UpdateApiDestination$ = [9, n0, _UAD,
    0, () => UpdateApiDestinationRequest$, () => UpdateApiDestinationResponse$
];
var UpdateArchive$ = [9, n0, _UA,
    0, () => UpdateArchiveRequest$, () => UpdateArchiveResponse$
];
var UpdateConnection$ = [9, n0, _UC,
    0, () => UpdateConnectionRequest$, () => UpdateConnectionResponse$
];
var UpdateEndpoint$ = [9, n0, _UE,
    0, () => UpdateEndpointRequest$, () => UpdateEndpointResponse$
];
var UpdateEventBus$ = [9, n0, _UEB,
    0, () => UpdateEventBusRequest$, () => UpdateEventBusResponse$
];

const getRuntimeConfig$1 = (config) => {
    return {
        apiVersion: "2015-10-07",
        base64Decoder: config?.base64Decoder ?? fromBase64,
        base64Encoder: config?.base64Encoder ?? toBase64,
        disableHostPrefix: config?.disableHostPrefix ?? false,
        endpointProvider: config?.endpointProvider ?? defaultEndpointResolver,
        extensions: config?.extensions ?? [],
        httpAuthSchemeProvider: config?.httpAuthSchemeProvider ?? defaultEventBridgeHttpAuthSchemeProvider,
        httpAuthSchemes: config?.httpAuthSchemes ?? [
            {
                schemeId: "aws.auth#sigv4",
                identityProvider: (ipc) => ipc.getIdentityProvider("aws.auth#sigv4"),
                signer: new AwsSdkSigV4Signer(),
            },
            {
                schemeId: "aws.auth#sigv4a",
                identityProvider: (ipc) => ipc.getIdentityProvider("aws.auth#sigv4a"),
                signer: new AwsSdkSigV4ASigner(),
            },
        ],
        logger: config?.logger ?? new NoOpLogger(),
        protocol: config?.protocol ?? AwsJson1_1Protocol,
        protocolSettings: config?.protocolSettings ?? {
            defaultNamespace: "com.amazonaws.eventbridge",
            errorTypeRegistries,
            xmlNamespace: "http://events.amazonaws.com/doc/2015-10-07",
            version: "2015-10-07",
            serviceTarget: "AWSEvents",
        },
        serviceId: config?.serviceId ?? "EventBridge",
        sha256: config?.sha256 ?? Sha256,
        signerConstructor: config?.signerConstructor ?? SignatureV4MultiRegion,
        urlParser: config?.urlParser ?? parseUrl,
        utf8Decoder: config?.utf8Decoder ?? fromUtf8,
        utf8Encoder: config?.utf8Encoder ?? toUtf8,
    };
};

const getRuntimeConfig = (config) => {
    emitWarningIfUnsupportedVersion(process.version);
    const defaultsMode = resolveDefaultsModeConfig(config);
    const defaultConfigProvider = () => defaultsMode().then(loadConfigsForDefaultMode);
    const clientSharedValues = getRuntimeConfig$1(config);
    emitWarningIfUnsupportedVersion$1(process.version);
    const loaderConfig = {
        profile: config?.profile,
        logger: clientSharedValues.logger,
    };
    return {
        ...clientSharedValues,
        ...config,
        runtime: "node",
        defaultsMode,
        authSchemePreference: config?.authSchemePreference ?? loadConfig(NODE_AUTH_SCHEME_PREFERENCE_OPTIONS, loaderConfig),
        bodyLengthChecker: config?.bodyLengthChecker ?? calculateBodyLength,
        credentialDefaultProvider: config?.credentialDefaultProvider ?? defaultProvider,
        defaultUserAgentProvider: config?.defaultUserAgentProvider ?? createDefaultUserAgentProvider({ serviceId: clientSharedValues.serviceId, clientVersion: packageInfo.version }),
        maxAttempts: config?.maxAttempts ?? loadConfig(NODE_MAX_ATTEMPT_CONFIG_OPTIONS, config),
        region: config?.region ?? loadConfig(NODE_REGION_CONFIG_OPTIONS, { ...NODE_REGION_CONFIG_FILE_OPTIONS, ...loaderConfig }),
        requestHandler: NodeHttpHandler.create(config?.requestHandler ?? defaultConfigProvider),
        retryMode: config?.retryMode ??
            loadConfig({
                ...NODE_RETRY_MODE_CONFIG_OPTIONS,
                default: async () => (await defaultConfigProvider()).retryMode || DEFAULT_RETRY_MODE,
            }, config),
        sigv4aSigningRegionSet: config?.sigv4aSigningRegionSet ?? loadConfig(NODE_SIGV4A_CONFIG_OPTIONS, loaderConfig),
        streamCollector: config?.streamCollector ?? streamCollector,
        useDualstackEndpoint: config?.useDualstackEndpoint ?? loadConfig(NODE_USE_DUALSTACK_ENDPOINT_CONFIG_OPTIONS, loaderConfig),
        useFipsEndpoint: config?.useFipsEndpoint ?? loadConfig(NODE_USE_FIPS_ENDPOINT_CONFIG_OPTIONS, loaderConfig),
        userAgentAppId: config?.userAgentAppId ?? loadConfig(NODE_APP_ID_CONFIG_OPTIONS, loaderConfig),
    };
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(getAwsRegionExtensionConfiguration(runtimeConfig), getDefaultExtensionConfiguration(runtimeConfig), getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, resolveAwsRegionExtensionConfiguration(extensionConfiguration), resolveDefaultRuntimeConfig(extensionConfiguration), resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class EventBridgeClient extends Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = resolveUserAgentConfig(_config_1);
        const _config_3 = resolveRetryConfig(_config_2);
        const _config_4 = resolveRegionConfig(_config_3);
        const _config_5 = resolveHostHeaderConfig(_config_4);
        const _config_6 = resolveEndpointConfig(_config_5);
        const _config_7 = resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(getUserAgentPlugin(this.config));
        this.middlewareStack.use(getRetryPlugin(this.config));
        this.middlewareStack.use(getContentLengthPlugin(this.config));
        this.middlewareStack.use(getHostHeaderPlugin(this.config));
        this.middlewareStack.use(getLoggerPlugin(this.config));
        this.middlewareStack.use(getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: defaultEventBridgeHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
                "aws.auth#sigv4a": config.credentials,
            }),
        }));
        this.middlewareStack.use(getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

const command = makeBuilder(commonParams, "AWSEvents", "EventBridgeClient", getEndpointPlugin);
const _ep0 = {};
const _ep1 = {
    EndpointId: { type: "contextParams", name: "EndpointId" },
};
const _mw0 = (Command, cs, config, o) => [];

class ActivateEventSourceCommand extends command(_ep0, _mw0, "ActivateEventSource", ActivateEventSource$) {
}

class CancelReplayCommand extends command(_ep0, _mw0, "CancelReplay", CancelReplay$) {
}

class CreateApiDestinationCommand extends command(_ep0, _mw0, "CreateApiDestination", CreateApiDestination$) {
}

class CreateArchiveCommand extends command(_ep0, _mw0, "CreateArchive", CreateArchive$) {
}

class CreateConnectionCommand extends command(_ep0, _mw0, "CreateConnection", CreateConnection$) {
}

class CreateEndpointCommand extends command(_ep0, _mw0, "CreateEndpoint", CreateEndpoint$) {
}

class CreateEventBusCommand extends command(_ep0, _mw0, "CreateEventBus", CreateEventBus$) {
}

class CreatePartnerEventSourceCommand extends command(_ep0, _mw0, "CreatePartnerEventSource", CreatePartnerEventSource$) {
}

class DeactivateEventSourceCommand extends command(_ep0, _mw0, "DeactivateEventSource", DeactivateEventSource$) {
}

class DeauthorizeConnectionCommand extends command(_ep0, _mw0, "DeauthorizeConnection", DeauthorizeConnection$) {
}

class DeleteApiDestinationCommand extends command(_ep0, _mw0, "DeleteApiDestination", DeleteApiDestination$) {
}

class DeleteArchiveCommand extends command(_ep0, _mw0, "DeleteArchive", DeleteArchive$) {
}

class DeleteConnectionCommand extends command(_ep0, _mw0, "DeleteConnection", DeleteConnection$) {
}

class DeleteEndpointCommand extends command(_ep0, _mw0, "DeleteEndpoint", DeleteEndpoint$) {
}

class DeleteEventBusCommand extends command(_ep0, _mw0, "DeleteEventBus", DeleteEventBus$) {
}

class DeletePartnerEventSourceCommand extends command(_ep0, _mw0, "DeletePartnerEventSource", DeletePartnerEventSource$) {
}

class DeleteRuleCommand extends command(_ep0, _mw0, "DeleteRule", DeleteRule$) {
}

class DescribeApiDestinationCommand extends command(_ep0, _mw0, "DescribeApiDestination", DescribeApiDestination$) {
}

class DescribeArchiveCommand extends command(_ep0, _mw0, "DescribeArchive", DescribeArchive$) {
}

class DescribeConnectionCommand extends command(_ep0, _mw0, "DescribeConnection", DescribeConnection$) {
}

class DescribeEndpointCommand extends command(_ep0, _mw0, "DescribeEndpoint", DescribeEndpoint$) {
}

class DescribeEventBusCommand extends command(_ep0, _mw0, "DescribeEventBus", DescribeEventBus$) {
}

class DescribeEventSourceCommand extends command(_ep0, _mw0, "DescribeEventSource", DescribeEventSource$) {
}

class DescribePartnerEventSourceCommand extends command(_ep0, _mw0, "DescribePartnerEventSource", DescribePartnerEventSource$) {
}

class DescribeReplayCommand extends command(_ep0, _mw0, "DescribeReplay", DescribeReplay$) {
}

class DescribeRuleCommand extends command(_ep0, _mw0, "DescribeRule", DescribeRule$) {
}

class DisableRuleCommand extends command(_ep0, _mw0, "DisableRule", DisableRule$) {
}

class EnableRuleCommand extends command(_ep0, _mw0, "EnableRule", EnableRule$) {
}

class ListApiDestinationsCommand extends command(_ep0, _mw0, "ListApiDestinations", ListApiDestinations$) {
}

class ListArchivesCommand extends command(_ep0, _mw0, "ListArchives", ListArchives$) {
}

class ListConnectionsCommand extends command(_ep0, _mw0, "ListConnections", ListConnections$) {
}

class ListEndpointsCommand extends command(_ep0, _mw0, "ListEndpoints", ListEndpoints$) {
}

class ListEventBusesCommand extends command(_ep0, _mw0, "ListEventBuses", ListEventBuses$) {
}

class ListEventSourcesCommand extends command(_ep0, _mw0, "ListEventSources", ListEventSources$) {
}

class ListPartnerEventSourceAccountsCommand extends command(_ep0, _mw0, "ListPartnerEventSourceAccounts", ListPartnerEventSourceAccounts$) {
}

class ListPartnerEventSourcesCommand extends command(_ep0, _mw0, "ListPartnerEventSources", ListPartnerEventSources$) {
}

class ListReplaysCommand extends command(_ep0, _mw0, "ListReplays", ListReplays$) {
}

class ListRuleNamesByTargetCommand extends command(_ep0, _mw0, "ListRuleNamesByTarget", ListRuleNamesByTarget$) {
}

class ListRulesCommand extends command(_ep0, _mw0, "ListRules", ListRules$) {
}

class ListTagsForResourceCommand extends command(_ep0, _mw0, "ListTagsForResource", ListTagsForResource$) {
}

class ListTargetsByRuleCommand extends command(_ep0, _mw0, "ListTargetsByRule", ListTargetsByRule$) {
}

class PutEventsCommand extends command(_ep1, _mw0, "PutEvents", PutEvents$) {
}

class PutPartnerEventsCommand extends command(_ep0, _mw0, "PutPartnerEvents", PutPartnerEvents$) {
}

class PutPermissionCommand extends command(_ep0, _mw0, "PutPermission", PutPermission$) {
}

class PutRuleCommand extends command(_ep0, _mw0, "PutRule", PutRule$) {
}

class PutTargetsCommand extends command(_ep0, _mw0, "PutTargets", PutTargets$) {
}

class RemovePermissionCommand extends command(_ep0, _mw0, "RemovePermission", RemovePermission$) {
}

class RemoveTargetsCommand extends command(_ep0, _mw0, "RemoveTargets", RemoveTargets$) {
}

class StartReplayCommand extends command(_ep0, _mw0, "StartReplay", StartReplay$) {
}

class TagResourceCommand extends command(_ep0, _mw0, "TagResource", TagResource$) {
}

class TestEventPatternCommand extends command(_ep0, _mw0, "TestEventPattern", TestEventPattern$) {
}

class UntagResourceCommand extends command(_ep0, _mw0, "UntagResource", UntagResource$) {
}

class UpdateApiDestinationCommand extends command(_ep0, _mw0, "UpdateApiDestination", UpdateApiDestination$) {
}

class UpdateArchiveCommand extends command(_ep0, _mw0, "UpdateArchive", UpdateArchive$) {
}

class UpdateConnectionCommand extends command(_ep0, _mw0, "UpdateConnection", UpdateConnection$) {
}

class UpdateEndpointCommand extends command(_ep0, _mw0, "UpdateEndpoint", UpdateEndpoint$) {
}

class UpdateEventBusCommand extends command(_ep0, _mw0, "UpdateEventBus", UpdateEventBus$) {
}

const commands = {
    ActivateEventSourceCommand,
    CancelReplayCommand,
    CreateApiDestinationCommand,
    CreateArchiveCommand,
    CreateConnectionCommand,
    CreateEndpointCommand,
    CreateEventBusCommand,
    CreatePartnerEventSourceCommand,
    DeactivateEventSourceCommand,
    DeauthorizeConnectionCommand,
    DeleteApiDestinationCommand,
    DeleteArchiveCommand,
    DeleteConnectionCommand,
    DeleteEndpointCommand,
    DeleteEventBusCommand,
    DeletePartnerEventSourceCommand,
    DeleteRuleCommand,
    DescribeApiDestinationCommand,
    DescribeArchiveCommand,
    DescribeConnectionCommand,
    DescribeEndpointCommand,
    DescribeEventBusCommand,
    DescribeEventSourceCommand,
    DescribePartnerEventSourceCommand,
    DescribeReplayCommand,
    DescribeRuleCommand,
    DisableRuleCommand,
    EnableRuleCommand,
    ListApiDestinationsCommand,
    ListArchivesCommand,
    ListConnectionsCommand,
    ListEndpointsCommand,
    ListEventBusesCommand,
    ListEventSourcesCommand,
    ListPartnerEventSourceAccountsCommand,
    ListPartnerEventSourcesCommand,
    ListReplaysCommand,
    ListRuleNamesByTargetCommand,
    ListRulesCommand,
    ListTagsForResourceCommand,
    ListTargetsByRuleCommand,
    PutEventsCommand,
    PutPartnerEventsCommand,
    PutPermissionCommand,
    PutRuleCommand,
    PutTargetsCommand,
    RemovePermissionCommand,
    RemoveTargetsCommand,
    StartReplayCommand,
    TagResourceCommand,
    TestEventPatternCommand,
    UntagResourceCommand,
    UpdateApiDestinationCommand,
    UpdateArchiveCommand,
    UpdateConnectionCommand,
    UpdateEndpointCommand,
    UpdateEventBusCommand,
};
class EventBridge extends EventBridgeClient {
}
createAggregatedClient(commands, EventBridge);

const ApiDestinationState = {
    ACTIVE: "ACTIVE",
    INACTIVE: "INACTIVE",
};
const ApiDestinationHttpMethod = {
    DELETE: "DELETE",
    GET: "GET",
    HEAD: "HEAD",
    OPTIONS: "OPTIONS",
    PATCH: "PATCH",
    POST: "POST",
    PUT: "PUT",
};
const ArchiveState = {
    CREATE_FAILED: "CREATE_FAILED",
    CREATING: "CREATING",
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
    UPDATE_FAILED: "UPDATE_FAILED",
    UPDATING: "UPDATING",
};
const AssignPublicIp = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const ReplayState = {
    CANCELLED: "CANCELLED",
    CANCELLING: "CANCELLING",
    COMPLETED: "COMPLETED",
    FAILED: "FAILED",
    RUNNING: "RUNNING",
    STARTING: "STARTING",
};
const ConnectionAuthorizationType = {
    API_KEY: "API_KEY",
    BASIC: "BASIC",
    OAUTH_CLIENT_CREDENTIALS: "OAUTH_CLIENT_CREDENTIALS",
};
const ConnectionOAuthHttpMethod = {
    GET: "GET",
    POST: "POST",
    PUT: "PUT",
};
const ConnectionState = {
    ACTIVE: "ACTIVE",
    AUTHORIZED: "AUTHORIZED",
    AUTHORIZING: "AUTHORIZING",
    CREATING: "CREATING",
    DEAUTHORIZED: "DEAUTHORIZED",
    DEAUTHORIZING: "DEAUTHORIZING",
    DELETING: "DELETING",
    FAILED_CONNECTIVITY: "FAILED_CONNECTIVITY",
    UPDATING: "UPDATING",
};
const ReplicationState = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const EndpointState = {
    ACTIVE: "ACTIVE",
    CREATE_FAILED: "CREATE_FAILED",
    CREATING: "CREATING",
    DELETE_FAILED: "DELETE_FAILED",
    DELETING: "DELETING",
    UPDATE_FAILED: "UPDATE_FAILED",
    UPDATING: "UPDATING",
};
const IncludeDetail = {
    FULL: "FULL",
    NONE: "NONE",
};
const Level = {
    ERROR: "ERROR",
    INFO: "INFO",
    OFF: "OFF",
    TRACE: "TRACE",
};
const EventSourceState = {
    ACTIVE: "ACTIVE",
    DELETED: "DELETED",
    PENDING: "PENDING",
};
const RuleState = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
    ENABLED_WITH_ALL_CLOUDTRAIL_MANAGEMENT_EVENTS: "ENABLED_WITH_ALL_CLOUDTRAIL_MANAGEMENT_EVENTS",
};
const LaunchType = {
    EC2: "EC2",
    EXTERNAL: "EXTERNAL",
    FARGATE: "FARGATE",
};
const PlacementConstraintType = {
    DISTINCT_INSTANCE: "distinctInstance",
    MEMBER_OF: "memberOf",
};
const PlacementStrategyType = {
    BINPACK: "binpack",
    RANDOM: "random",
    SPREAD: "spread",
};
const PropagateTags = {
    TASK_DEFINITION: "TASK_DEFINITION",
};

exports.AccessDeniedException = AccessDeniedException;
exports.AccessDeniedException$ = AccessDeniedException$;
exports.ActivateEventSource$ = ActivateEventSource$;
exports.ActivateEventSourceCommand = ActivateEventSourceCommand;
exports.ActivateEventSourceRequest$ = ActivateEventSourceRequest$;
exports.ApiDestination$ = ApiDestination$;
exports.ApiDestinationHttpMethod = ApiDestinationHttpMethod;
exports.ApiDestinationState = ApiDestinationState;
exports.AppSyncParameters$ = AppSyncParameters$;
exports.Archive$ = Archive$;
exports.ArchiveState = ArchiveState;
exports.AssignPublicIp = AssignPublicIp;
exports.AwsVpcConfiguration$ = AwsVpcConfiguration$;
exports.BatchArrayProperties$ = BatchArrayProperties$;
exports.BatchParameters$ = BatchParameters$;
exports.BatchRetryStrategy$ = BatchRetryStrategy$;
exports.CancelReplay$ = CancelReplay$;
exports.CancelReplayCommand = CancelReplayCommand;
exports.CancelReplayRequest$ = CancelReplayRequest$;
exports.CancelReplayResponse$ = CancelReplayResponse$;
exports.CapacityProviderStrategyItem$ = CapacityProviderStrategyItem$;
exports.ConcurrentModificationException = ConcurrentModificationException;
exports.ConcurrentModificationException$ = ConcurrentModificationException$;
exports.Condition$ = Condition$;
exports.Connection$ = Connection$;
exports.ConnectionApiKeyAuthResponseParameters$ = ConnectionApiKeyAuthResponseParameters$;
exports.ConnectionAuthResponseParameters$ = ConnectionAuthResponseParameters$;
exports.ConnectionAuthorizationType = ConnectionAuthorizationType;
exports.ConnectionBasicAuthResponseParameters$ = ConnectionBasicAuthResponseParameters$;
exports.ConnectionBodyParameter$ = ConnectionBodyParameter$;
exports.ConnectionHeaderParameter$ = ConnectionHeaderParameter$;
exports.ConnectionHttpParameters$ = ConnectionHttpParameters$;
exports.ConnectionOAuthClientResponseParameters$ = ConnectionOAuthClientResponseParameters$;
exports.ConnectionOAuthHttpMethod = ConnectionOAuthHttpMethod;
exports.ConnectionOAuthResponseParameters$ = ConnectionOAuthResponseParameters$;
exports.ConnectionQueryStringParameter$ = ConnectionQueryStringParameter$;
exports.ConnectionState = ConnectionState;
exports.ConnectivityResourceConfigurationArn$ = ConnectivityResourceConfigurationArn$;
exports.ConnectivityResourceParameters$ = ConnectivityResourceParameters$;
exports.CreateApiDestination$ = CreateApiDestination$;
exports.CreateApiDestinationCommand = CreateApiDestinationCommand;
exports.CreateApiDestinationRequest$ = CreateApiDestinationRequest$;
exports.CreateApiDestinationResponse$ = CreateApiDestinationResponse$;
exports.CreateArchive$ = CreateArchive$;
exports.CreateArchiveCommand = CreateArchiveCommand;
exports.CreateArchiveRequest$ = CreateArchiveRequest$;
exports.CreateArchiveResponse$ = CreateArchiveResponse$;
exports.CreateConnection$ = CreateConnection$;
exports.CreateConnectionApiKeyAuthRequestParameters$ = CreateConnectionApiKeyAuthRequestParameters$;
exports.CreateConnectionAuthRequestParameters$ = CreateConnectionAuthRequestParameters$;
exports.CreateConnectionBasicAuthRequestParameters$ = CreateConnectionBasicAuthRequestParameters$;
exports.CreateConnectionCommand = CreateConnectionCommand;
exports.CreateConnectionOAuthClientRequestParameters$ = CreateConnectionOAuthClientRequestParameters$;
exports.CreateConnectionOAuthRequestParameters$ = CreateConnectionOAuthRequestParameters$;
exports.CreateConnectionRequest$ = CreateConnectionRequest$;
exports.CreateConnectionResponse$ = CreateConnectionResponse$;
exports.CreateEndpoint$ = CreateEndpoint$;
exports.CreateEndpointCommand = CreateEndpointCommand;
exports.CreateEndpointRequest$ = CreateEndpointRequest$;
exports.CreateEndpointResponse$ = CreateEndpointResponse$;
exports.CreateEventBus$ = CreateEventBus$;
exports.CreateEventBusCommand = CreateEventBusCommand;
exports.CreateEventBusRequest$ = CreateEventBusRequest$;
exports.CreateEventBusResponse$ = CreateEventBusResponse$;
exports.CreatePartnerEventSource$ = CreatePartnerEventSource$;
exports.CreatePartnerEventSourceCommand = CreatePartnerEventSourceCommand;
exports.CreatePartnerEventSourceRequest$ = CreatePartnerEventSourceRequest$;
exports.CreatePartnerEventSourceResponse$ = CreatePartnerEventSourceResponse$;
exports.DeactivateEventSource$ = DeactivateEventSource$;
exports.DeactivateEventSourceCommand = DeactivateEventSourceCommand;
exports.DeactivateEventSourceRequest$ = DeactivateEventSourceRequest$;
exports.DeadLetterConfig$ = DeadLetterConfig$;
exports.DeauthorizeConnection$ = DeauthorizeConnection$;
exports.DeauthorizeConnectionCommand = DeauthorizeConnectionCommand;
exports.DeauthorizeConnectionRequest$ = DeauthorizeConnectionRequest$;
exports.DeauthorizeConnectionResponse$ = DeauthorizeConnectionResponse$;
exports.DeleteApiDestination$ = DeleteApiDestination$;
exports.DeleteApiDestinationCommand = DeleteApiDestinationCommand;
exports.DeleteApiDestinationRequest$ = DeleteApiDestinationRequest$;
exports.DeleteApiDestinationResponse$ = DeleteApiDestinationResponse$;
exports.DeleteArchive$ = DeleteArchive$;
exports.DeleteArchiveCommand = DeleteArchiveCommand;
exports.DeleteArchiveRequest$ = DeleteArchiveRequest$;
exports.DeleteArchiveResponse$ = DeleteArchiveResponse$;
exports.DeleteConnection$ = DeleteConnection$;
exports.DeleteConnectionCommand = DeleteConnectionCommand;
exports.DeleteConnectionRequest$ = DeleteConnectionRequest$;
exports.DeleteConnectionResponse$ = DeleteConnectionResponse$;
exports.DeleteEndpoint$ = DeleteEndpoint$;
exports.DeleteEndpointCommand = DeleteEndpointCommand;
exports.DeleteEndpointRequest$ = DeleteEndpointRequest$;
exports.DeleteEndpointResponse$ = DeleteEndpointResponse$;
exports.DeleteEventBus$ = DeleteEventBus$;
exports.DeleteEventBusCommand = DeleteEventBusCommand;
exports.DeleteEventBusRequest$ = DeleteEventBusRequest$;
exports.DeletePartnerEventSource$ = DeletePartnerEventSource$;
exports.DeletePartnerEventSourceCommand = DeletePartnerEventSourceCommand;
exports.DeletePartnerEventSourceRequest$ = DeletePartnerEventSourceRequest$;
exports.DeleteRule$ = DeleteRule$;
exports.DeleteRuleCommand = DeleteRuleCommand;
exports.DeleteRuleRequest$ = DeleteRuleRequest$;
exports.DescribeApiDestination$ = DescribeApiDestination$;
exports.DescribeApiDestinationCommand = DescribeApiDestinationCommand;
exports.DescribeApiDestinationRequest$ = DescribeApiDestinationRequest$;
exports.DescribeApiDestinationResponse$ = DescribeApiDestinationResponse$;
exports.DescribeArchive$ = DescribeArchive$;
exports.DescribeArchiveCommand = DescribeArchiveCommand;
exports.DescribeArchiveRequest$ = DescribeArchiveRequest$;
exports.DescribeArchiveResponse$ = DescribeArchiveResponse$;
exports.DescribeConnection$ = DescribeConnection$;
exports.DescribeConnectionCommand = DescribeConnectionCommand;
exports.DescribeConnectionConnectivityParameters$ = DescribeConnectionConnectivityParameters$;
exports.DescribeConnectionRequest$ = DescribeConnectionRequest$;
exports.DescribeConnectionResourceParameters$ = DescribeConnectionResourceParameters$;
exports.DescribeConnectionResponse$ = DescribeConnectionResponse$;
exports.DescribeEndpoint$ = DescribeEndpoint$;
exports.DescribeEndpointCommand = DescribeEndpointCommand;
exports.DescribeEndpointRequest$ = DescribeEndpointRequest$;
exports.DescribeEndpointResponse$ = DescribeEndpointResponse$;
exports.DescribeEventBus$ = DescribeEventBus$;
exports.DescribeEventBusCommand = DescribeEventBusCommand;
exports.DescribeEventBusRequest$ = DescribeEventBusRequest$;
exports.DescribeEventBusResponse$ = DescribeEventBusResponse$;
exports.DescribeEventSource$ = DescribeEventSource$;
exports.DescribeEventSourceCommand = DescribeEventSourceCommand;
exports.DescribeEventSourceRequest$ = DescribeEventSourceRequest$;
exports.DescribeEventSourceResponse$ = DescribeEventSourceResponse$;
exports.DescribePartnerEventSource$ = DescribePartnerEventSource$;
exports.DescribePartnerEventSourceCommand = DescribePartnerEventSourceCommand;
exports.DescribePartnerEventSourceRequest$ = DescribePartnerEventSourceRequest$;
exports.DescribePartnerEventSourceResponse$ = DescribePartnerEventSourceResponse$;
exports.DescribeReplay$ = DescribeReplay$;
exports.DescribeReplayCommand = DescribeReplayCommand;
exports.DescribeReplayRequest$ = DescribeReplayRequest$;
exports.DescribeReplayResponse$ = DescribeReplayResponse$;
exports.DescribeRule$ = DescribeRule$;
exports.DescribeRuleCommand = DescribeRuleCommand;
exports.DescribeRuleRequest$ = DescribeRuleRequest$;
exports.DescribeRuleResponse$ = DescribeRuleResponse$;
exports.DisableRule$ = DisableRule$;
exports.DisableRuleCommand = DisableRuleCommand;
exports.DisableRuleRequest$ = DisableRuleRequest$;
exports.EcsParameters$ = EcsParameters$;
exports.EnableRule$ = EnableRule$;
exports.EnableRuleCommand = EnableRuleCommand;
exports.EnableRuleRequest$ = EnableRuleRequest$;
exports.Endpoint$ = Endpoint$;
exports.EndpointEventBus$ = EndpointEventBus$;
exports.EndpointState = EndpointState;
exports.EventBridge = EventBridge;
exports.EventBridgeClient = EventBridgeClient;
exports.EventBridgeServiceException = EventBridgeServiceException;
exports.EventBridgeServiceException$ = EventBridgeServiceException$;
exports.EventBus$ = EventBus$;
exports.EventSource$ = EventSource$;
exports.EventSourceState = EventSourceState;
exports.FailoverConfig$ = FailoverConfig$;
exports.HttpParameters$ = HttpParameters$;
exports.IllegalStatusException = IllegalStatusException;
exports.IllegalStatusException$ = IllegalStatusException$;
exports.IncludeDetail = IncludeDetail;
exports.InputTransformer$ = InputTransformer$;
exports.InternalException = InternalException;
exports.InternalException$ = InternalException$;
exports.InvalidEventPatternException = InvalidEventPatternException;
exports.InvalidEventPatternException$ = InvalidEventPatternException$;
exports.InvalidStateException = InvalidStateException;
exports.InvalidStateException$ = InvalidStateException$;
exports.KinesisParameters$ = KinesisParameters$;
exports.LaunchType = LaunchType;
exports.Level = Level;
exports.LimitExceededException = LimitExceededException;
exports.LimitExceededException$ = LimitExceededException$;
exports.ListApiDestinations$ = ListApiDestinations$;
exports.ListApiDestinationsCommand = ListApiDestinationsCommand;
exports.ListApiDestinationsRequest$ = ListApiDestinationsRequest$;
exports.ListApiDestinationsResponse$ = ListApiDestinationsResponse$;
exports.ListArchives$ = ListArchives$;
exports.ListArchivesCommand = ListArchivesCommand;
exports.ListArchivesRequest$ = ListArchivesRequest$;
exports.ListArchivesResponse$ = ListArchivesResponse$;
exports.ListConnections$ = ListConnections$;
exports.ListConnectionsCommand = ListConnectionsCommand;
exports.ListConnectionsRequest$ = ListConnectionsRequest$;
exports.ListConnectionsResponse$ = ListConnectionsResponse$;
exports.ListEndpoints$ = ListEndpoints$;
exports.ListEndpointsCommand = ListEndpointsCommand;
exports.ListEndpointsRequest$ = ListEndpointsRequest$;
exports.ListEndpointsResponse$ = ListEndpointsResponse$;
exports.ListEventBuses$ = ListEventBuses$;
exports.ListEventBusesCommand = ListEventBusesCommand;
exports.ListEventBusesRequest$ = ListEventBusesRequest$;
exports.ListEventBusesResponse$ = ListEventBusesResponse$;
exports.ListEventSources$ = ListEventSources$;
exports.ListEventSourcesCommand = ListEventSourcesCommand;
exports.ListEventSourcesRequest$ = ListEventSourcesRequest$;
exports.ListEventSourcesResponse$ = ListEventSourcesResponse$;
exports.ListPartnerEventSourceAccounts$ = ListPartnerEventSourceAccounts$;
exports.ListPartnerEventSourceAccountsCommand = ListPartnerEventSourceAccountsCommand;
exports.ListPartnerEventSourceAccountsRequest$ = ListPartnerEventSourceAccountsRequest$;
exports.ListPartnerEventSourceAccountsResponse$ = ListPartnerEventSourceAccountsResponse$;
exports.ListPartnerEventSources$ = ListPartnerEventSources$;
exports.ListPartnerEventSourcesCommand = ListPartnerEventSourcesCommand;
exports.ListPartnerEventSourcesRequest$ = ListPartnerEventSourcesRequest$;
exports.ListPartnerEventSourcesResponse$ = ListPartnerEventSourcesResponse$;
exports.ListReplays$ = ListReplays$;
exports.ListReplaysCommand = ListReplaysCommand;
exports.ListReplaysRequest$ = ListReplaysRequest$;
exports.ListReplaysResponse$ = ListReplaysResponse$;
exports.ListRuleNamesByTarget$ = ListRuleNamesByTarget$;
exports.ListRuleNamesByTargetCommand = ListRuleNamesByTargetCommand;
exports.ListRuleNamesByTargetRequest$ = ListRuleNamesByTargetRequest$;
exports.ListRuleNamesByTargetResponse$ = ListRuleNamesByTargetResponse$;
exports.ListRules$ = ListRules$;
exports.ListRulesCommand = ListRulesCommand;
exports.ListRulesRequest$ = ListRulesRequest$;
exports.ListRulesResponse$ = ListRulesResponse$;
exports.ListTagsForResource$ = ListTagsForResource$;
exports.ListTagsForResourceCommand = ListTagsForResourceCommand;
exports.ListTagsForResourceRequest$ = ListTagsForResourceRequest$;
exports.ListTagsForResourceResponse$ = ListTagsForResourceResponse$;
exports.ListTargetsByRule$ = ListTargetsByRule$;
exports.ListTargetsByRuleCommand = ListTargetsByRuleCommand;
exports.ListTargetsByRuleRequest$ = ListTargetsByRuleRequest$;
exports.ListTargetsByRuleResponse$ = ListTargetsByRuleResponse$;
exports.LogConfig$ = LogConfig$;
exports.ManagedRuleException = ManagedRuleException;
exports.ManagedRuleException$ = ManagedRuleException$;
exports.NetworkConfiguration$ = NetworkConfiguration$;
exports.OperationDisabledException = OperationDisabledException;
exports.OperationDisabledException$ = OperationDisabledException$;
exports.PartnerEventSource$ = PartnerEventSource$;
exports.PartnerEventSourceAccount$ = PartnerEventSourceAccount$;
exports.PlacementConstraint$ = PlacementConstraint$;
exports.PlacementConstraintType = PlacementConstraintType;
exports.PlacementStrategy$ = PlacementStrategy$;
exports.PlacementStrategyType = PlacementStrategyType;
exports.PolicyLengthExceededException = PolicyLengthExceededException;
exports.PolicyLengthExceededException$ = PolicyLengthExceededException$;
exports.Primary$ = Primary$;
exports.PropagateTags = PropagateTags;
exports.PutEvents$ = PutEvents$;
exports.PutEventsCommand = PutEventsCommand;
exports.PutEventsRequest$ = PutEventsRequest$;
exports.PutEventsRequestEntry$ = PutEventsRequestEntry$;
exports.PutEventsResponse$ = PutEventsResponse$;
exports.PutEventsResultEntry$ = PutEventsResultEntry$;
exports.PutPartnerEvents$ = PutPartnerEvents$;
exports.PutPartnerEventsCommand = PutPartnerEventsCommand;
exports.PutPartnerEventsRequest$ = PutPartnerEventsRequest$;
exports.PutPartnerEventsRequestEntry$ = PutPartnerEventsRequestEntry$;
exports.PutPartnerEventsResponse$ = PutPartnerEventsResponse$;
exports.PutPartnerEventsResultEntry$ = PutPartnerEventsResultEntry$;
exports.PutPermission$ = PutPermission$;
exports.PutPermissionCommand = PutPermissionCommand;
exports.PutPermissionRequest$ = PutPermissionRequest$;
exports.PutRule$ = PutRule$;
exports.PutRuleCommand = PutRuleCommand;
exports.PutRuleRequest$ = PutRuleRequest$;
exports.PutRuleResponse$ = PutRuleResponse$;
exports.PutTargets$ = PutTargets$;
exports.PutTargetsCommand = PutTargetsCommand;
exports.PutTargetsRequest$ = PutTargetsRequest$;
exports.PutTargetsResponse$ = PutTargetsResponse$;
exports.PutTargetsResultEntry$ = PutTargetsResultEntry$;
exports.RedshiftDataParameters$ = RedshiftDataParameters$;
exports.RemovePermission$ = RemovePermission$;
exports.RemovePermissionCommand = RemovePermissionCommand;
exports.RemovePermissionRequest$ = RemovePermissionRequest$;
exports.RemoveTargets$ = RemoveTargets$;
exports.RemoveTargetsCommand = RemoveTargetsCommand;
exports.RemoveTargetsRequest$ = RemoveTargetsRequest$;
exports.RemoveTargetsResponse$ = RemoveTargetsResponse$;
exports.RemoveTargetsResultEntry$ = RemoveTargetsResultEntry$;
exports.Replay$ = Replay$;
exports.ReplayDestination$ = ReplayDestination$;
exports.ReplayState = ReplayState;
exports.ReplicationConfig$ = ReplicationConfig$;
exports.ReplicationState = ReplicationState;
exports.ResourceAlreadyExistsException = ResourceAlreadyExistsException;
exports.ResourceAlreadyExistsException$ = ResourceAlreadyExistsException$;
exports.ResourceNotFoundException = ResourceNotFoundException;
exports.ResourceNotFoundException$ = ResourceNotFoundException$;
exports.RetryPolicy$ = RetryPolicy$;
exports.RoutingConfig$ = RoutingConfig$;
exports.Rule$ = Rule$;
exports.RuleState = RuleState;
exports.RunCommandParameters$ = RunCommandParameters$;
exports.RunCommandTarget$ = RunCommandTarget$;
exports.SageMakerPipelineParameter$ = SageMakerPipelineParameter$;
exports.SageMakerPipelineParameters$ = SageMakerPipelineParameters$;
exports.Secondary$ = Secondary$;
exports.SqsParameters$ = SqsParameters$;
exports.StartReplay$ = StartReplay$;
exports.StartReplayCommand = StartReplayCommand;
exports.StartReplayRequest$ = StartReplayRequest$;
exports.StartReplayResponse$ = StartReplayResponse$;
exports.Tag$ = Tag$;
exports.TagResource$ = TagResource$;
exports.TagResourceCommand = TagResourceCommand;
exports.TagResourceRequest$ = TagResourceRequest$;
exports.TagResourceResponse$ = TagResourceResponse$;
exports.Target$ = Target$;
exports.TestEventPattern$ = TestEventPattern$;
exports.TestEventPatternCommand = TestEventPatternCommand;
exports.TestEventPatternRequest$ = TestEventPatternRequest$;
exports.TestEventPatternResponse$ = TestEventPatternResponse$;
exports.ThrottlingException = ThrottlingException;
exports.ThrottlingException$ = ThrottlingException$;
exports.UntagResource$ = UntagResource$;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UntagResourceRequest$ = UntagResourceRequest$;
exports.UntagResourceResponse$ = UntagResourceResponse$;
exports.UpdateApiDestination$ = UpdateApiDestination$;
exports.UpdateApiDestinationCommand = UpdateApiDestinationCommand;
exports.UpdateApiDestinationRequest$ = UpdateApiDestinationRequest$;
exports.UpdateApiDestinationResponse$ = UpdateApiDestinationResponse$;
exports.UpdateArchive$ = UpdateArchive$;
exports.UpdateArchiveCommand = UpdateArchiveCommand;
exports.UpdateArchiveRequest$ = UpdateArchiveRequest$;
exports.UpdateArchiveResponse$ = UpdateArchiveResponse$;
exports.UpdateConnection$ = UpdateConnection$;
exports.UpdateConnectionApiKeyAuthRequestParameters$ = UpdateConnectionApiKeyAuthRequestParameters$;
exports.UpdateConnectionAuthRequestParameters$ = UpdateConnectionAuthRequestParameters$;
exports.UpdateConnectionBasicAuthRequestParameters$ = UpdateConnectionBasicAuthRequestParameters$;
exports.UpdateConnectionCommand = UpdateConnectionCommand;
exports.UpdateConnectionOAuthClientRequestParameters$ = UpdateConnectionOAuthClientRequestParameters$;
exports.UpdateConnectionOAuthRequestParameters$ = UpdateConnectionOAuthRequestParameters$;
exports.UpdateConnectionRequest$ = UpdateConnectionRequest$;
exports.UpdateConnectionResponse$ = UpdateConnectionResponse$;
exports.UpdateEndpoint$ = UpdateEndpoint$;
exports.UpdateEndpointCommand = UpdateEndpointCommand;
exports.UpdateEndpointRequest$ = UpdateEndpointRequest$;
exports.UpdateEndpointResponse$ = UpdateEndpointResponse$;
exports.UpdateEventBus$ = UpdateEventBus$;
exports.UpdateEventBusCommand = UpdateEventBusCommand;
exports.UpdateEventBusRequest$ = UpdateEventBusRequest$;
exports.UpdateEventBusResponse$ = UpdateEventBusResponse$;
exports.errorTypeRegistries = errorTypeRegistries;
